<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Part 4 Basic RNA-Seq Analysis – From fastq files to DE genes | Introductory Bioinformatics for VAI Users</title>
  <meta name="description" content="Bioinformatics tutorials with hands-on exercises and examples." />
  <meta name="generator" content="bookdown 0.36 and GitBook 2.6.7" />

  <meta property="og:title" content="Part 4 Basic RNA-Seq Analysis – From fastq files to DE genes | Introductory Bioinformatics for VAI Users" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Bioinformatics tutorials with hands-on exercises and examples." />
  <meta name="github-repo" content="vari-bbc/Intro_to_Linux_for_HPC" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Part 4 Basic RNA-Seq Analysis – From fastq files to DE genes | Introductory Bioinformatics for VAI Users" />
  
  <meta name="twitter:description" content="Bioinformatics tutorials with hands-on exercises and examples." />
  

<meta name="author" content="The Bioinformatics and Biostatistics Core (BBC)" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="mini-project.html"/>
<link rel="next" href="pathway-enrichment-and-experimental-design.html"/>
<script src="assets/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="assets/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="assets/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="assets/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="assets/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="assets/kePrint-0.0.1/kePrint.js"></script>
<link href="assets/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> <strong>Introduction</strong></a></li>
<li class="chapter" data-level="2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html"><i class="fa fa-check"></i><b>2</b> <strong>Basic Command Line Utilities on the HPC</strong></a>
<ul>
<li class="chapter" data-level="2.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#access-hpc"><i class="fa fa-check"></i><b>2.1</b> <strong>Access HPC</strong></a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#ondemand-instructions"><i class="fa fa-check"></i><b>2.1.1</b> <strong>OnDemand Instructions</strong></a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#file-navigation"><i class="fa fa-check"></i><b>2.2</b> <strong>File navigation</strong></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#navigate-to-a-directory-folder"><i class="fa fa-check"></i><b>2.2.1</b> <strong>Navigate to a directory (folder)</strong></a></li>
<li class="chapter" data-level="2.2.2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#list-content-in-a-directory"><i class="fa fa-check"></i><b>2.2.2</b> <strong>List content in a directory</strong></a></li>
<li class="chapter" data-level="2.2.3" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#navigate-back-to-home-directory"><i class="fa fa-check"></i><b>2.2.3</b> <strong>Navigate back to home directory</strong></a></li>
<li class="chapter" data-level="2.2.4" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#create-a-directory"><i class="fa fa-check"></i><b>2.2.4</b> <strong>Create a directory</strong></a></li>
<li class="chapter" data-level="2.2.5" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#copy-a-file"><i class="fa fa-check"></i><b>2.2.5</b> <strong>Copy a file</strong></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#view-and-manipulate-files"><i class="fa fa-check"></i><b>2.3</b> <strong>View and manipulate files</strong></a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#display-content"><i class="fa fa-check"></i><b>2.3.1</b> <strong>Display content</strong></a></li>
<li class="chapter" data-level="2.3.2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#pattern-search"><i class="fa fa-check"></i><b>2.3.2</b> <strong>Pattern search</strong></a></li>
<li class="chapter" data-level="2.3.3" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#display-the-number-of-lines-words-and-characters"><i class="fa fa-check"></i><b>2.3.3</b> <strong>Display the number of lines, words, and characters</strong></a></li>
<li class="chapter" data-level="2.3.4" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#pipe---redirection"><i class="fa fa-check"></i><b>2.3.4</b> <strong>Pipe - redirection</strong></a></li>
<li class="chapter" data-level="2.3.5" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#output-redirection"><i class="fa fa-check"></i><b>2.3.5</b> <strong>Output redirection</strong></a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#exercise"><i class="fa fa-check"></i><b>2.4</b> <strong>Exercise</strong></a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#step-1-copy-the-files"><i class="fa fa-check"></i><b>2.4.1</b> <strong>Step 1: Copy the files</strong></a></li>
<li class="chapter" data-level="2.4.2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#step-2-combine-the-files"><i class="fa fa-check"></i><b>2.4.2</b> <strong>Step 2: Combine the files</strong></a></li>
<li class="chapter" data-level="2.4.3" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#step-3-count-the-number-of-reads"><i class="fa fa-check"></i><b>2.4.3</b> <strong>Step 3: Count the number of reads</strong></a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#summary"><i class="fa fa-check"></i><b>2.5</b> <strong>Summary</strong></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="mini-project.html"><a href="mini-project.html"><i class="fa fa-check"></i><b>3</b> <strong>Mini Project</strong></a>
<ul>
<li class="chapter" data-level="3.1" data-path="mini-project.html"><a href="mini-project.html#start-an-interactive-job"><i class="fa fa-check"></i><b>3.1</b> <strong>Start an Interactive Job</strong></a></li>
<li class="chapter" data-level="3.2" data-path="mini-project.html"><a href="mini-project.html#create-directory"><i class="fa fa-check"></i><b>3.2</b> <strong>Create Directory</strong></a></li>
<li class="chapter" data-level="3.3" data-path="mini-project.html"><a href="mini-project.html#copy-fastq-files-to-fastqs-subdirectory"><i class="fa fa-check"></i><b>3.3</b> <strong>Copy Fastq Files To <code>fastqs</code> subdirectory</strong></a></li>
<li class="chapter" data-level="3.4" data-path="mini-project.html"><a href="mini-project.html#use-basic-linux-commands-to-explore-the-fastq-files"><i class="fa fa-check"></i><b>3.4</b> <strong>Use basic Linux commands to explore the fastq files</strong></a></li>
<li class="chapter" data-level="3.5" data-path="mini-project.html"><a href="mini-project.html#working-with-environment-modules"><i class="fa fa-check"></i><b>3.5</b> <strong>Working With Environment Modules</strong></a></li>
<li class="chapter" data-level="3.6" data-path="mini-project.html"><a href="mini-project.html#run-fastqc"><i class="fa fa-check"></i><b>3.6</b> <strong>Run FastQC</strong></a></li>
<li class="chapter" data-level="3.7" data-path="mini-project.html"><a href="mini-project.html#set-up-a-job-script-to-run-salmon-to-align-the-reads"><i class="fa fa-check"></i><b>3.7</b> <strong>Set up a job script to run Salmon to align the reads</strong></a></li>
<li class="chapter" data-level="3.8" data-path="mini-project.html"><a href="mini-project.html#submit-a-job"><i class="fa fa-check"></i><b>3.8</b> <strong>Submit a Job</strong></a></li>
<li class="chapter" data-level="3.9" data-path="mini-project.html"><a href="mini-project.html#examine-the-standard-output-stdout-and-standard-error-stderr-log-files"><i class="fa fa-check"></i><b>3.9</b> <strong>Examine the Standard Output (stdout) and Standard Error (stderr) log files</strong></a></li>
<li class="chapter" data-level="3.10" data-path="mini-project.html"><a href="mini-project.html#use-grep-to-find-the-tpms-for-a-specific-gene"><i class="fa fa-check"></i><b>3.10</b> <strong>Use grep to find the TPMs for a specific gene</strong></a></li>
<li class="chapter" data-level="3.11" data-path="mini-project.html"><a href="mini-project.html#bonus-use-an-interactive-job-to-run-multiqc-on-the-salmon-and-fastqc-output"><i class="fa fa-check"></i><b>3.11</b> <strong>BONUS: Use an interactive job to run multiQC on the Salmon and FastQC output</strong></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><i class="fa fa-check"></i><b>4</b> <strong>Basic RNA-Seq Analysis – From fastq files to DE genes</strong></a>
<ul>
<li class="chapter" data-level="4.1" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#run-the-bbc-pipeline-to-align-the-reads"><i class="fa fa-check"></i><b>4.1</b> <strong>Run the BBC pipeline to align the reads</strong></a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#the-workflow-as-a-graph"><i class="fa fa-check"></i><b>4.1.1</b> <strong>The workflow as a graph</strong></a></li>
<li class="chapter" data-level="4.1.2" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#clone-the-github-repo"><i class="fa fa-check"></i><b>4.1.2</b> <strong>Clone the Github repo</strong></a></li>
<li class="chapter" data-level="4.1.3" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#add-your-fastq-files-to-raw_data"><i class="fa fa-check"></i><b>4.1.3</b> <strong>Add your fastq files to <code>raw_data/</code></strong></a></li>
<li class="chapter" data-level="4.1.4" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#fill-out-the-samplesheet"><i class="fa fa-check"></i><b>4.1.4</b> <strong>Fill out the samplesheet</strong></a></li>
<li class="chapter" data-level="4.1.5" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#fill-out-the-config-file"><i class="fa fa-check"></i><b>4.1.5</b> <strong>Fill out the config file</strong></a></li>
<li class="chapter" data-level="4.1.6" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#submit-the-main-snakemake-job"><i class="fa fa-check"></i><b>4.1.6</b> <strong>Submit the main Snakemake job</strong></a></li>
<li class="chapter" data-level="4.1.7" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#bbc-maintained-reference-files"><i class="fa fa-check"></i><b>4.1.7</b> <strong>BBC-maintained reference files</strong></a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#quality-control"><i class="fa fa-check"></i><b>4.2</b> <strong>Quality control</strong></a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#multiqc-report"><i class="fa fa-check"></i><b>4.2.1</b> <strong>multiQC report</strong></a></li>
<li class="chapter" data-level="4.2.2" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#other-considerations"><i class="fa fa-check"></i><b>4.2.2</b> <strong>Other considerations</strong></a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#exploring-the-data-using-isee"><i class="fa fa-check"></i><b>4.3</b> <strong>Exploring the data using iSEE</strong></a></li>
<li class="chapter" data-level="4.4" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#de-analysis-using-r-and-deseq2"><i class="fa fa-check"></i><b>4.4</b> <strong>DE analysis using R and DESeq2</strong></a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#make-an-output-directory"><i class="fa fa-check"></i><b>4.4.1</b> <strong>Make an output directory</strong></a></li>
<li class="chapter" data-level="4.4.2" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#load-packages"><i class="fa fa-check"></i><b>4.4.2</b> <strong>Load packages</strong></a></li>
<li class="chapter" data-level="4.4.3" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#set-up-your-deseq-object"><i class="fa fa-check"></i><b>4.4.3</b> <strong>Set up your DESeq object</strong></a></li>
<li class="chapter" data-level="4.4.4" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#remove-genes-with-lowno-expression"><i class="fa fa-check"></i><b>4.4.4</b> <strong>Remove genes with low/no expression</strong></a></li>
<li class="chapter" data-level="4.4.5" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#different-normalization-approaches-for-different-biases"><i class="fa fa-check"></i><b>4.4.5</b> <strong>Different normalization approaches for different biases</strong></a></li>
<li class="chapter" data-level="4.4.6" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#run-the-de-workflow"><i class="fa fa-check"></i><b>4.4.6</b> <strong>Run the DE workflow</strong></a></li>
<li class="chapter" data-level="4.4.7" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#summarize-de-results"><i class="fa fa-check"></i><b>4.4.7</b> <strong>Summarize DE results</strong></a></li>
<li class="chapter" data-level="4.4.8" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#shrink-log-fold-changes-for-lowly-expressed-genes"><i class="fa fa-check"></i><b>4.4.8</b> <strong>Shrink log fold changes for lowly expressed genes</strong></a></li>
<li class="chapter" data-level="4.4.9" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#output-de-results"><i class="fa fa-check"></i><b>4.4.9</b> <strong>Output DE results</strong></a></li>
<li class="chapter" data-level="4.4.10" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#look-for-specific-genes"><i class="fa fa-check"></i><b>4.4.10</b> <strong>Look for specific genes</strong></a></li>
<li class="chapter" data-level="4.4.11" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#output-tables-with-raw-counts"><i class="fa fa-check"></i><b>4.4.11</b> <strong>Output tables with raw counts</strong></a></li>
<li class="chapter" data-level="4.4.12" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#output-tables-with-log2-normalized-counts"><i class="fa fa-check"></i><b>4.4.12</b> <strong>Output tables with log2 normalized counts</strong></a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#some-common-plots-for-deg-analysis"><i class="fa fa-check"></i><b>4.5</b> <strong>Some common plots for DEG analysis</strong></a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#volcano-plot"><i class="fa fa-check"></i><b>4.5.1</b> <strong>Volcano plot</strong></a></li>
<li class="chapter" data-level="4.5.2" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#heatmap"><i class="fa fa-check"></i><b>4.5.2</b> <strong>Heatmap</strong></a></li>
<li class="chapter" data-level="4.5.3" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#p-value-distribution"><i class="fa fa-check"></i><b>4.5.3</b> <strong>P value distribution</strong></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="pathway-enrichment-and-experimental-design.html"><a href="pathway-enrichment-and-experimental-design.html"><i class="fa fa-check"></i><b>5</b> <strong>Pathway, enrichment, and experimental design</strong></a>
<ul>
<li class="chapter" data-level="5.1" data-path="pathway-enrichment-and-experimental-design.html"><a href="pathway-enrichment-and-experimental-design.html#downstream-of-de-gene-analysis"><i class="fa fa-check"></i><b>5.1</b> <strong>Downstream of DE gene analysis</strong></a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="pathway-enrichment-and-experimental-design.html"><a href="pathway-enrichment-and-experimental-design.html#copy-the-project-directory"><i class="fa fa-check"></i><b>5.1.1</b> <strong>Copy the project directory</strong></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="appendices.html"><a href="appendices.html"><i class="fa fa-check"></i><b>6</b> <strong>Appendices</strong></a>
<ul>
<li class="chapter" data-level="6.1" data-path="appendices.html"><a href="appendices.html#workshop-powerpoint-files"><i class="fa fa-check"></i><b>6.1</b> <strong>Workshop Powerpoint files</strong></a></li>
<li class="chapter" data-level="6.2" data-path="appendices.html"><a href="appendices.html#local-access-of-files-on-the-hpc"><i class="fa fa-check"></i><b>6.2</b> <strong>Local access of files on the HPC</strong></a></li>
<li class="chapter" data-level="6.3" data-path="appendices.html"><a href="appendices.html#bash-cheatsheet"><i class="fa fa-check"></i><b>6.3</b> <strong>Bash cheatsheet</strong></a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introductory Bioinformatics for VAI Users</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="basic-rna-seq-analysis-from-fastq-files-to-de-genes" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Part 4</span> <strong>Basic RNA-Seq Analysis – From fastq files to DE genes</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#basic-rna-seq-analysis-from-fastq-files-to-de-genes" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="run-the-bbc-pipeline-to-align-the-reads" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> <strong>Run the BBC pipeline to align the reads</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#run-the-bbc-pipeline-to-align-the-reads" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="the-workflow-as-a-graph" class="section level3 hasAnchor" number="4.1.1">
<h3><span class="header-section-number">4.1.1</span> <strong>The workflow as a graph</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#the-workflow-as-a-graph" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="float">
<img src="img/rulegraph.png" alt="The workflow." />
<div class="figcaption">The workflow.</div>
</div>
</div>
<div id="clone-the-github-repo" class="section level3 hasAnchor" number="4.1.2">
<h3><span class="header-section-number">4.1.2</span> <strong>Clone the Github repo</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#clone-the-github-repo" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Instructions for running it are on the <a href="https://github.com/vari-bbc/rnaseq_workflow">Github README</a>, but here we will go through each step in more detail.</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb71-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb71-1" tabindex="-1"></a><span class="bu">cd</span> /varidata/researchtemp/hpctmp/BBC_workshop_June2023_II/</span>
<span id="cb71-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb71-2" tabindex="-1"></a></span>
<span id="cb71-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb71-3" tabindex="-1"></a><span class="fu">mkdir</span> <span class="op">&lt;</span>username<span class="op">&gt;</span></span>
<span id="cb71-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb71-4" tabindex="-1"></a></span>
<span id="cb71-5"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb71-5" tabindex="-1"></a><span class="bu">cd</span> <span class="op">&lt;</span>username<span class="op">&gt;</span></span></code></pre></div>
<div class="sourceCode" id="cb72"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb72-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb72-1" tabindex="-1"></a><span class="fu">git</span> clone https://github.com/vari-bbc/rnaseq_workflow.git </span></code></pre></div>
<p>The Github repository consisting of the RNA-seq workflow should now be downloaded in a folder named <code>rnaseq_workflow</code>.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb73-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb73-1" tabindex="-1"></a><span class="bu">cd</span> rnaseq_workflow</span>
<span id="cb73-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb73-2" tabindex="-1"></a></span>
<span id="cb73-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb73-3" tabindex="-1"></a><span class="fu">ls</span></span></code></pre></div>
</div>
<div id="add-your-fastq-files-to-raw_data" class="section level3 hasAnchor" number="4.1.3">
<h3><span class="header-section-number">4.1.3</span> <strong>Add your fastq files to <code>raw_data/</code></strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#add-your-fastq-files-to-raw_data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Instead of making multiple copies of the same file, which can quickly use up your lab’s storage quota, we can use <strong>symbolic links</strong>. The sequence data that we will be using for this workshop are from the <a href="https://pubmed.ncbi.nlm.nih.gov/24926665/">airway dataset</a> referenced in the DESeq2 vignette. The gene counts can actually be downloaded as an <a href="https://bioconductor.org/packages/release/data/experiment/html/airway.html">R package</a>.</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb74-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb74-1" tabindex="-1"></a><span class="fu">ls</span> ../../0_fastqs/</span></code></pre></div>
<div class="sourceCode" id="cb75"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb75-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb75-1" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-sr</span> ../../0_fastqs/<span class="pp">*</span> ./raw_data</span></code></pre></div>
<p>You can see where the symbolic links are pointing to using <code>ls -l</code>.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb76-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb76-1" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-l</span> ./raw_data</span></code></pre></div>
</div>
<div id="fill-out-the-samplesheet" class="section level3 hasAnchor" number="4.1.4">
<h3><span class="header-section-number">4.1.4</span> <strong>Fill out the samplesheet</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#fill-out-the-samplesheet" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The samplesheet is a tab-delimited file within <code>config/samplesheet/</code> and is named <code>units.tsv</code>. The easiest way to fill this out is to run the helper script, <code>make_units_template.sh</code>, to generate a template, then edit using a text editor.</p>
<p>FOR THIS WORKSHOP, NO CHANGES NEED TO BE MADE.</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb77-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb77-1" tabindex="-1"></a><span class="bu">cd</span> config/samplesheet/</span>
<span id="cb77-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb77-2" tabindex="-1"></a></span>
<span id="cb77-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb77-3" tabindex="-1"></a><span class="ex">./make_units_template.sh</span></span></code></pre></div>
<p>There should now be a file named ‘units_template.tsv’.</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb78-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb78-1" tabindex="-1"></a><span class="fu">ls</span></span></code></pre></div>
<p>We can replace the ‘units.tsv’ with ‘units_template.tsv’.</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb79-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb79-1" tabindex="-1"></a><span class="fu">mv</span> units_template.tsv units.tsv</span></code></pre></div>
<p>Go back to the base level of the project directory.</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb80-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb80-1" tabindex="-1"></a><span class="bu">cd</span> ../..</span></code></pre></div>
<p>Make sure you are at <code>/varidata/researchtemp/hpctmp/BBC_workshop_June2023_II/&lt;username&gt;/rnaseq_workflow</code>.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb81-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb81-1" tabindex="-1"></a><span class="bu">pwd</span> </span></code></pre></div>
</div>
<div id="fill-out-the-config-file" class="section level3 hasAnchor" number="4.1.5">
<h3><span class="header-section-number">4.1.5</span> <strong>Fill out the config file</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#fill-out-the-config-file" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The config file is a <a href="https://en.wikipedia.org/wiki/YAML">YAML</a> file indicating the locations of reference files and also contains options for the workflow that you can turn off or turn on.</p>
<p>Typically, the main thing is to specify reference files corresponding to the species you have sequenced (human, mouse, or rat etc). For this workshop, we are dealing with human data so we will align to the hg38 reference.</p>
<p>Index files allow alignment algorithms to align reads to specific reference sequences.</p>
<p>FOR THIS WORKSHOP, NO CHANGES NEED TO BE MADE.</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb82-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb82-1" tabindex="-1"></a></span>
<span id="cb82-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb82-2" tabindex="-1"></a><span class="fu">cat</span> config/config.yaml</span></code></pre></div>
</div>
<div id="submit-the-main-snakemake-job" class="section level3 hasAnchor" number="4.1.6">
<h3><span class="header-section-number">4.1.6</span> <strong>Submit the main Snakemake job</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#submit-the-main-snakemake-job" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb83"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb83-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb83-1" tabindex="-1"></a><span class="ex">sbatch</span> <span class="at">-p</span> big bin/run_snake.sh</span></code></pre></div>
</div>
<div id="bbc-maintained-reference-files" class="section level3 hasAnchor" number="4.1.7">
<h3><span class="header-section-number">4.1.7</span> <strong>BBC-maintained reference files</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#bbc-maintained-reference-files" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For future reference, the BBC downloads and maintains commonly used files and indexes for several model species. These files are version controlled to promote reproducibility in case you need to rerun an analysis or you want to run the exact same analysis on different datasets.</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb84-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb84-1" tabindex="-1"></a><span class="fu">ls</span> /varidata/research/projects/bbc/versioned_references/</span></code></pre></div>
<pre><code>## 2021-04-09_12.44.55_v1
## 2021-07-02_11.23.48_v3
## 2021-07-13_09.57.31_v4
## 2021-08-02_20.40.24_v5
## 2021-08-10_11.12.27_v6
## 2021-10-25_15.31.38_v7
## 2022-01-26_13.39.07_v8
## 2022-03-08_14.47.50_v9
## 2022-10-06_14.25.40_v10
## 2022-12-19_13.27.04_v11
## 2023-05-03_15.28.41_v12
## 2023-10-04_10.07.51_v13
## 2023-10-06_11.13.48_v14
## 2023-11-08_15.45.32_v15
## 2023-11-09_09.38.05_v16
## latest
## regarding_v2.txt</code></pre>
<div class="sourceCode" id="cb86"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb86-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb86-1" tabindex="-1"></a><span class="fu">ls</span> /varidata/research/projects/bbc/versioned_references/2023-05-03_15.28.41_v12/data/</span></code></pre></div>
<pre><code>## GRCz11
## c.elegans-WBcel235
## dm6_BDGP6.28.100
## e.coli-K12-mg1665_ensembl
## hg19_gencode
## hg19_gencode_plus_ERCC92
## hg19_gencode_plus_viruses
## hg38_gencode
## mm10_gencode
## mm10_gencode_plus_ERCC92
## mm10_gencode_plus_e.coli-K12-mg1665_ensembl
## mm10_gencode_plus_viruses_and_cfmedips
## mm11_gencode
## mm9_ucsc
## mm_BALB_CJ
## mm_BALB_CJ_plus_viruses_and_cfmedips
## mm_FVB_NJ
## mm_FVB_NJ_plus_viruses_and_cfmedips
## rnor6_ensembl</code></pre>
<p>The source of these files can be found in the <code>species.tsv</code> file.</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb88-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb88-1" tabindex="-1"></a><span class="fu">cat</span> /varidata/research/projects/bbc/versioned_references/2023-05-03_15.28.41_v12/bin/species.tsv  <span class="kw">|</span> <span class="fu">cut</span> <span class="at">-f2-3</span></span></code></pre></div>
<pre><code>## id   genome_fasta
## mm10_gencode ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M24/GRCm38.primary_assembly.genome.fa.gz
## hg38_gencode ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_33/GRCh38.primary_assembly.genome.fa.gz
## e.coli-K12-mg1665_ensembl    ftp://ftp.ensemblgenomes.org/pub/release-47/bacteria//fasta/bacteria_0_collection/escherichia_coli_str_k_12_substr_mg1655/dna/Escherichia_coli_str_k_12_substr_mg1655.ASM584v2.dna.chromosome.Chromosome.fa.gz
## hg19_gencode ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_36/GRCh37_mapping/GRCh37.primary_assembly.genome.fa.gz
## dm6_BDGP6.28.100 ftp://ftp.ensembl.org/pub/release-100/fasta/drosophila_melanogaster/dna/Drosophila_melanogaster.BDGP6.28.dna.toplevel.fa.gz
## c.elegans-WBcel235   http://ftp.ensembl.org/pub/release-103/fasta/caenorhabditis_elegans/dna/Caenorhabditis_elegans.WBcel235.dna.toplevel.fa.gz
## mm_BALB_CJ   http://ftp.ensembl.org/pub/release-104/fasta/mus_musculus_balbcj/dna/Mus_musculus_balbcj.BALB_cJ_v1.dna.toplevel.fa.gz
## mm_FVB_NJ    http://ftp.ensembl.org/pub/release-104/fasta/mus_musculus_fvbnj/dna/Mus_musculus_fvbnj.FVB_NJ_v1.dna.toplevel.fa.gz
## rnor6_ensembl    http://ftp.ensembl.org/pub/release-104/fasta/rattus_norvegicus/dna/Rattus_norvegicus.Rnor_6.0.dna.toplevel.fa.gz
## GRCz11   https://ftp.ensembl.org/pub/release-108/fasta/danio_rerio/dna/Danio_rerio.GRCz11.dna.toplevel.fa.gz
## mm11_gencode https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M31/GRCm39.primary_assembly.genome.fa.gz
## mm9_ucsc ftp://hgdownload.cse.ucsc.edu/goldenPath/mm9/bigZips/mm9.fa.gz</code></pre>
<p>The verions of the software used to generate the index files can be found in the <code>config.yaml</code> file. Sometimes index files are not backwards-compatible, meaning index files generated by a newer version of a tool cannot be used by an older version of the tool.</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb90-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb90-1" tabindex="-1"></a><span class="fu">cat</span> /varidata/research/projects/bbc/versioned_references/2023-05-03_15.28.41_v12/bin/config.yaml</span></code></pre></div>
<pre><code>## # This is the directory that the rsync rule will use as the source directory. It shuold be the run directory this workflow.
## sourceDir: &quot;/varidata/research/projects/bbc/research/prep_bbc_shared_current&quot;
## 
## # This is where the timestamped directories and the &#39;latest&#39; symlink will be created.
## timestamp_dir: &quot;/varidata/research/projects/bbc/versioned_references/&quot;
## 
## # Environment modules used.
## # Specify all environment modules here instead of within the Snakefile itself to make it easy to look over quickly.
## # When aligning using the index files created by this workflow, use the same version aligner as listed here to ensure compatibility.
## samtools:   &quot;bbc/samtools/samtools-1.9&quot;
## picard:     &quot;bbc/picard/picard-2.21.4-SNAPSHOT&quot;
## STAR:       &quot;bbc/STAR/STAR-2.7.8a&quot;
## biscuit:    &quot;bbc/biscuit/biscuit_1_0_1&quot;
## bwa:        &quot;bbc/bwa/bwa-0.7.17&quot;
## bowtie2:    &quot;bbc/bowtie2/bowtie2-2.4.1&quot;
## python3:    &quot;bbc/python3/python-3.8.1&quot;
## bismark:    &quot;bbc/bismark/bismark-0.23.0&quot;
## kb-python:  &quot;bbc/kb-python/kb-python-0.24.4&quot;
## parallel:   &quot;bbc/parallel/parallel-20191122&quot;
## gsutil:     &quot;bbc/gsutil/gsutil-4.52&quot;
## seqtk:      &quot;bbc/seqtk/seqtk-1.3-r115-dirty&quot;
## salmon:     &quot;bbc/salmon/salmon-1.4.0&quot;
## kallisto:   &quot;bbc/kallisto/kallisto-0.46.1&quot;</code></pre>
</div>
</div>
<div id="quality-control" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> <strong>Quality control</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#quality-control" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>It’s important to look through QC metrics to ensure that the data is likely to produce meaningful results. Ideally, you don’t want to waste time trying to interpret bad data.</p>
<div id="multiqc-report" class="section level3 hasAnchor" number="4.2.1">
<h3><span class="header-section-number">4.2.1</span> <strong>multiQC report</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#multiqc-report" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Navigate to your project folder using Finder (Mac) or File Explorer (Windows). Find <code>results/multiqc/multiqc_report.html</code> and double-click it.</p>
<ul>
<li>Alignment rate?</li>
<li>Higher duplication rate can be normal for RNA-seq.</li>
<li>Strandedness; In our workflow, this is automatically inferred using Salmon.</li>
<li>Any contamination? Check FastqScreen results.</li>
</ul>
</div>
<div id="other-considerations" class="section level3 hasAnchor" number="4.2.2">
<h3><span class="header-section-number">4.2.2</span> <strong>Other considerations</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#other-considerations" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Not a bad idea to look quickly at the alignments in IGV. Any sign of gDNA contamination?</li>
<li>Mutations in specific genotypes? Is it exonic? Can do a quick check in IGV.</li>
<li>How many genes expressed? Easy to check in R.</li>
<li>Marker genes? Can do a quick check in iSEE if you expect certain expression patterns between groups.</li>
<li>Any genes supposed to knocked out or knocked down? Use iSEE.</li>
<li>Does the PCA show clustering? Use iSEE.</li>
</ul>
</div>
</div>
<div id="exploring-the-data-using-isee" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> <strong>Exploring the data using iSEE</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#exploring-the-data-using-isee" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Navigate to your project folder using Finder (Mac) or File Explorer (Windows). Find <code>iSEE/app.R</code> and double-click it.</p>
<div class="float">
<img src="img/iSEE_screenshot.png" alt="iSEE screenshot." />
<div class="figcaption">iSEE screenshot.</div>
</div>
</div>
<div id="de-analysis-using-r-and-deseq2" class="section level2 hasAnchor" number="4.4">
<h2><span class="header-section-number">4.4</span> <strong>DE analysis using R and DESeq2</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#de-analysis-using-r-and-deseq2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="make-an-output-directory" class="section level3 hasAnchor" number="4.4.1">
<h3><span class="header-section-number">4.4.1</span> <strong>Make an output directory</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#make-an-output-directory" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb92-1" tabindex="-1"></a>outdir <span class="ot">&lt;-</span> <span class="st">&quot;./deseq2_out_files/&quot;</span></span>
<span id="cb92-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb92-2" tabindex="-1"></a></span>
<span id="cb92-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb92-3" tabindex="-1"></a><span class="fu">dir.create</span>(outdir, <span class="at">recursive=</span><span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="load-packages" class="section level3 hasAnchor" number="4.4.2">
<h3><span class="header-section-number">4.4.2</span> <strong>Load packages</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#load-packages" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb93-1" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(dplyr))</span>
<span id="cb93-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb93-2" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb93-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb93-3" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb93-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb93-4" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb93-5"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb93-5" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb93-6"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb93-6" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(ComplexHeatmap))</span>
<span id="cb93-7"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb93-7" tabindex="-1"></a><span class="fu">suppressMessages</span>(<span class="fu">library</span>(DESeq2))</span></code></pre></div>
</div>
<div id="set-up-your-deseq-object" class="section level3 hasAnchor" number="4.4.3">
<h3><span class="header-section-number">4.4.3</span> <strong>Set up your DESeq object</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#set-up-your-deseq-object" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb94-1" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../results/SummarizedExperiment/SummarizedExperiment.rds&quot;</span>)</span></code></pre></div>
<p>Let’s take a look to see what assays are stored in the SummarizedExperiment object. Note that DESeq2 assumes the first assay is the raw counts.</p>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb95-1" tabindex="-1"></a><span class="fu">assayNames</span>(se)</span></code></pre></div>
<pre><code>## [1] &quot;counts&quot; &quot;tpms&quot;   &quot;vst&quot;</code></pre>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb97-1" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">assayNames</span>(se)[<span class="dv">1</span>] <span class="sc">==</span> <span class="st">&quot;counts&quot;</span>)</span></code></pre></div>
<p>To print more information about this SummarizedExperiment object, you can just type its name.</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb98-1" tabindex="-1"></a>se</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 60721 8 
## metadata(0):
## assays(3): counts tpms vst
## rownames(60721): ENSG00000223972.5 ENSG00000227232.5 ...
##   ENSG00000278625.1 ENSG00000277374.1
## rowData names(4): Symbol Uniq_syms entrez Gene_name
## colnames(8): SRR1039509 SRR1039513 ... SRR1039516 SRR1039520
## colData names(2): sample group</code></pre>
<p>The counts and the meta data need to be stored inside a special DESeq2 object called a ‘DESeqDataSet’. Here, we also specify that each gene will be fit with a model design of ‘~ group’.</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb100-1" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSet</span>(se, <span class="at">design =</span> <span class="sc">~</span> group)</span></code></pre></div>
<pre><code>## converting counts to integer mode</code></pre>
<pre><code>## Warning in DESeqDataSet(se, design = ~group): some variables in design formula
## are characters, converting to factors</code></pre>
</div>
<div id="remove-genes-with-lowno-expression" class="section level3 hasAnchor" number="4.4.4">
<h3><span class="header-section-number">4.4.4</span> <strong>Remove genes with low/no expression</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#remove-genes-with-lowno-expression" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We cannot do meaningful analyses of genes with very low counts. This will speed up the analysis.</p>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb103-1" tabindex="-1"></a><span class="co"># prefilter genes, keeping only genes with 10 or more total read counts across samples</span></span>
<span id="cb103-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb103-2" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(dds)) <span class="sc">&gt;=</span> <span class="dv">10</span></span>
<span id="cb103-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb103-3" tabindex="-1"></a></span>
<span id="cb103-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb103-4" tabindex="-1"></a><span class="fu">message</span>(<span class="fu">str_glue</span>(<span class="st">&quot;Keeping {sum(keep)} genes.&quot;</span>))</span></code></pre></div>
<pre><code>## Keeping 21485 genes.</code></pre>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb105-1" tabindex="-1"></a>dds <span class="ot">&lt;-</span> dds[keep, ]</span></code></pre></div>
</div>
<div id="different-normalization-approaches-for-different-biases" class="section level3 hasAnchor" number="4.4.5">
<h3><span class="header-section-number">4.4.5</span> <strong>Different normalization approaches for different biases</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#different-normalization-approaches-for-different-biases" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><a href="https://vanandelinstitute-my.sharepoint.com/:b:/g/personal/kin_lau_vai_org/EcruvwL-OrBIvCzXZ7HMPlcBo65fu0pucrivMmCwzM98dA?e=yCkfTa">Types of biases in RNA-seq</a></p>
</div>
<div id="run-the-de-workflow" class="section level3 hasAnchor" number="4.4.6">
<h3><span class="header-section-number">4.4.6</span> <strong>Run the DE workflow</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#run-the-de-workflow" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <a href="https://www.rdocumentation.org/packages/DESeq2/versions/1.12.3/topics/DESeq">DESeq</a> function is a convenience function from DESeq2 that estimates size factors (normalization) and fits negative binomial GLMs.</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb106-1" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span></code></pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb113-1" tabindex="-1"></a><span class="fu">message</span>(<span class="fu">paste0</span>(<span class="st">&quot;Coefficient names are: &quot;</span>, <span class="fu">paste</span>(<span class="fu">resultsNames</span>(dds), <span class="at">collapse =</span> <span class="st">&quot; &quot;</span>)))</span></code></pre></div>
<pre><code>## Coefficient names are: Intercept group_untrt_vs_trt</code></pre>
<p>After the models are fitted, we can test specific pairs of groups for differential expression. For DESeq2, it is recommended to provide the significance cutoff that you wish to use as it affects the multiple testing correction procedure (see <a href="https://www.rdocumentation.org/packages/DESeq2/versions/1.12.3/topics/results">docs</a>).</p>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb115-1" tabindex="-1"></a>contrast <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;group&quot;</span>, <span class="st">&quot;trt&quot;</span>, <span class="st">&quot;untrt&quot;</span>)</span>
<span id="cb115-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb115-2" tabindex="-1"></a>fdr_cutoff <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb115-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb115-3" tabindex="-1"></a></span>
<span id="cb115-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb115-4" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast=</span>contrast, <span class="at">alpha=</span>fdr_cutoff)</span>
<span id="cb115-5"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb115-5" tabindex="-1"></a>res <span class="ot">&lt;-</span> res[<span class="fu">order</span>(res<span class="sc">$</span>pvalue), ]</span></code></pre></div>
</div>
<div id="summarize-de-results" class="section level3 hasAnchor" number="4.4.7">
<h3><span class="header-section-number">4.4.7</span> <strong>Summarize DE results</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#summarize-de-results" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb116-1" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(res)</span>
<span id="cb116-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb116-2" tabindex="-1"></a><span class="fu">data.frame</span>(</span>
<span id="cb116-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb116-3" tabindex="-1"></a>  <span class="at">UP=</span><span class="fu">sum</span>(df<span class="sc">$</span>padj <span class="sc">&lt;=</span> fdr_cutoff <span class="sc">&amp;</span> df<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb116-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb116-4" tabindex="-1"></a>  <span class="at">DWN=</span><span class="fu">sum</span>(df<span class="sc">$</span>padj <span class="sc">&lt;=</span> fdr_cutoff <span class="sc">&amp;</span> df<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb116-5"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb116-5" tabindex="-1"></a>  <span class="at">Tested=</span><span class="fu">sum</span>(<span class="sc">!</span><span class="fu">is.na</span>(df<span class="sc">$</span>padj))</span>
<span id="cb116-6"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb116-6" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>##     UP  DWN Tested
## 1 1872 1481  17272</code></pre>
</div>
<div id="shrink-log-fold-changes-for-lowly-expressed-genes" class="section level3 hasAnchor" number="4.4.8">
<h3><span class="header-section-number">4.4.8</span> <strong>Shrink log fold changes for lowly expressed genes</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#shrink-log-fold-changes-for-lowly-expressed-genes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This step does not affect the identification of DE genes, but it can be useful to perform this to obtain more reliable estimates of the log fold changes for visualizations or for ranking genes (e.g. GSEA).</p>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb118-1" tabindex="-1"></a>lfc_shrink <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds, <span class="at">contrast=</span>contrast, <span class="at">type=</span><span class="st">&quot;ashr&quot;</span>)</span></code></pre></div>
<pre><code>## using &#39;ashr&#39; for LFC shrinkage. If used in published research, please cite:
##     Stephens, M. (2016) False discovery rates: a new deal. Biostatistics, 18:2.
##     https://doi.org/10.1093/biostatistics/kxw041</code></pre>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb120-1" tabindex="-1"></a>lfc_shrink <span class="ot">&lt;-</span> lfc_shrink[<span class="fu">order</span>(lfc_shrink<span class="sc">$</span>pvalue), ]</span></code></pre></div>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb121-1" tabindex="-1"></a>DESeq2<span class="sc">::</span><span class="fu">plotMA</span>(res, <span class="at">main=</span><span class="st">&quot;Default LFC&quot;</span>)</span></code></pre></div>
<p><img src="03_rnaseq_workflow_files/figure-html/ma_plots-1.png" width="480" /></p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb122-1" tabindex="-1"></a>DESeq2<span class="sc">::</span><span class="fu">plotMA</span>(lfc_shrink, <span class="at">main=</span><span class="st">&quot;Shrunken LFC&quot;</span>)</span></code></pre></div>
<p><img src="03_rnaseq_workflow_files/figure-html/ma_plots_shrunken-1.png" width="480" /></p>
</div>
<div id="output-de-results" class="section level3 hasAnchor" number="4.4.9">
<h3><span class="header-section-number">4.4.9</span> <strong>Output DE results</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#output-de-results" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Here, we merge the different gene name columns to the DE results and output to a tab-delimited file, which can be opened in Excel for manual perusal.</p>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb123-1" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">as.data.frame</span>(<span class="fu">rowData</span>(dds)[<span class="fu">rownames</span>(lfc_shrink), <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]),</span>
<span id="cb123-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb123-2" tabindex="-1"></a>            <span class="fu">as.data.frame</span>(lfc_shrink)) <span class="sc">%&gt;%</span></span>
<span id="cb123-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb123-3" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;ens_gene&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb124-1" tabindex="-1"></a><span class="fu">write_tsv</span>(df, <span class="fu">file.path</span>(outdir, <span class="st">&quot;de_res.tsv&quot;</span>))</span></code></pre></div>
</div>
<div id="look-for-specific-genes" class="section level3 hasAnchor" number="4.4.10">
<h3><span class="header-section-number">4.4.10</span> <strong>Look for specific genes</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#look-for-specific-genes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We know certain genes should be differentially expressed based on the <a href="https://pubmed.ncbi.nlm.nih.gov/24926665/">paper</a> that this dataset came from. We can check that these genes were significantly DE in our analysis.</p>
<p>Likewise, this would be a good time to check for knocked down, knocked out etc genes if such prior knowledge is available, though that is not always the case.</p>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb125-1" tabindex="-1"></a>df <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(Symbol <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;DUSP1&quot;</span>, <span class="st">&quot;KLF15&quot;</span>, <span class="st">&quot;CRISPLD2&quot;</span>))</span></code></pre></div>
<pre><code>##             ens_gene   Symbol Uniq_syms entrez
## 1  ENSG00000120129.6    DUSP1     DUSP1   1843
## 2  ENSG00000163884.4    KLF15     KLF15  28999
## 3 ENSG00000103196.12 CRISPLD2  CRISPLD2  83716
##                                                  Gene_name  baseMean
## 1                           dual specificity phosphatase 1 3357.1543
## 2                              KLF transcription factor 15  551.1765
## 3 cysteine rich secretory protein LCCL domain containing 2 2875.4571
##   log2FoldChange     lfcSE       pvalue         padj
## 1       2.899715 0.1982447 9.251009e-50 3.195669e-46
## 2       4.325185 0.4142736 2.361153e-28 1.568532e-25
## 3       2.618796 0.2590196 2.407565e-25 1.144831e-22</code></pre>
</div>
<div id="output-tables-with-raw-counts" class="section level3 hasAnchor" number="4.4.11">
<h3><span class="header-section-number">4.4.11</span> <strong>Output tables with raw counts</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#output-tables-with-raw-counts" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Some folks also find it useful to have tables of the raw counts or the normalized counts. The raw counts can be extracted from the DESeq2 object using either <code>assay()</code> or <code>counts()</code>.</p>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb127-1" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">as.data.frame</span>(<span class="fu">rowData</span>(dds)[, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]),</span>
<span id="cb127-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb127-2" tabindex="-1"></a>            <span class="fu">assay</span>(dds, <span class="st">&quot;counts&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb127-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb127-3" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;ens_gene&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb128-1" tabindex="-1"></a><span class="fu">write_tsv</span>(df, <span class="fu">file.path</span>(outdir, <span class="st">&quot;counts.tsv&quot;</span>))</span></code></pre></div>
</div>
<div id="output-tables-with-log2-normalized-counts" class="section level3 hasAnchor" number="4.4.12">
<h3><span class="header-section-number">4.4.12</span> <strong>Output tables with log2 normalized counts</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#output-tables-with-log2-normalized-counts" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For the log2 normalized counts, we commonly use the variance stabilized transformation (<a href="https://rdrr.io/bioc/DESeq2/man/varianceStabilizingTransformation.html">VST</a>). These values can be used for heatmaps, clustering or other downstream applications.</p>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb129-1" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind=</span><span class="cn">FALSE</span>)</span>
<span id="cb129-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb129-2" tabindex="-1"></a></span>
<span id="cb129-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb129-3" tabindex="-1"></a>vst_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">cbind</span>(<span class="fu">rowData</span>(vsd)[, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>], <span class="fu">assay</span>(vsd))) <span class="sc">%&gt;%</span></span>
<span id="cb129-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb129-4" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;ens_gene&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb130-1" tabindex="-1"></a><span class="fu">write_tsv</span>(vst_df, <span class="fu">file.path</span>(outdir, <span class="st">&quot;vst.tsv&quot;</span>))</span></code></pre></div>
</div>
</div>
<div id="some-common-plots-for-deg-analysis" class="section level2 hasAnchor" number="4.5">
<h2><span class="header-section-number">4.5</span> <strong>Some common plots for DEG analysis</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#some-common-plots-for-deg-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="volcano-plot" class="section level3 hasAnchor" number="4.5.1">
<h3><span class="header-section-number">4.5.1</span> <strong>Volcano plot</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#volcano-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-1" tabindex="-1"></a>make_volcano <span class="ot">&lt;-</span> <span class="cf">function</span>(df, pval_nm, <span class="at">pval_cutoff=</span><span class="fl">0.1</span>){</span>
<span id="cb131-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-2" tabindex="-1"></a>  <span class="co"># remove genes with NA for pvalue</span></span>
<span id="cb131-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-3" tabindex="-1"></a>  df <span class="ot">&lt;-</span> df[<span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(df[[pval_nm]])), ]</span>
<span id="cb131-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-4" tabindex="-1"></a>  </span>
<span id="cb131-5"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-5" tabindex="-1"></a>  <span class="co"># add gene names</span></span>
<span id="cb131-6"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-6" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(df, <span class="fu">rowData</span>(dds)[<span class="fu">rownames</span>(df), <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>])</span>
<span id="cb131-7"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-7" tabindex="-1"></a>  </span>
<span id="cb131-8"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-8" tabindex="-1"></a>  top_genes <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb131-9"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-9" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="fu">abs</span>(df<span class="sc">$</span>log2FoldChange))) <span class="sc">%&gt;%</span></span>
<span id="cb131-10"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-10" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">row_number</span>() <span class="sc">&lt;=</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb131-11"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-11" tabindex="-1"></a>    <span class="fu">rownames</span>()</span>
<span id="cb131-12"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-12" tabindex="-1"></a>  </span>
<span id="cb131-13"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-13" tabindex="-1"></a>  df<span class="sc">$</span>Sig <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(df<span class="sc">$</span>padj <span class="sc">&lt;=</span> pval_cutoff, <span class="st">&quot;Sig&quot;</span>, <span class="st">&quot;NS&quot;</span>)</span>
<span id="cb131-14"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-14" tabindex="-1"></a>  </span>
<span id="cb131-15"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-15" tabindex="-1"></a>  df[[pval_nm]] <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(df[[pval_nm]])</span>
<span id="cb131-16"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-16" tabindex="-1"></a>  </span>
<span id="cb131-17"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-17" tabindex="-1"></a>  </span>
<span id="cb131-18"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-18" tabindex="-1"></a>  <span class="fu">ggplot</span>(df, <span class="fu">aes_string</span>(<span class="at">x=</span><span class="st">&quot;log2FoldChange&quot;</span>, <span class="at">y=</span>pval_nm)) <span class="sc">+</span></span>
<span id="cb131-19"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-19" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color=</span>Sig), <span class="at">size=</span><span class="fl">0.6</span>) <span class="sc">+</span></span>
<span id="cb131-20"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-20" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;black&quot;</span>, <span class="st">&quot;salmon&quot;</span>)) <span class="sc">+</span></span>
<span id="cb131-21"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-21" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">ylab</span>(<span class="fu">str_glue</span>(<span class="st">&quot;-log10(&quot;</span>, pval_nm,<span class="st">&quot;)&quot;</span>)) <span class="sc">+</span></span>
<span id="cb131-22"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-22" tabindex="-1"></a>    <span class="fu">geom_text_repel</span>(<span class="at">data=</span>df[top_genes, ],</span>
<span id="cb131-23"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-23" tabindex="-1"></a>                    <span class="fu">aes</span>(<span class="at">label=</span>Uniq_syms), <span class="at">max.overlaps=</span><span class="cn">Inf</span>, <span class="at">min.segment.length =</span> <span class="dv">0</span>)</span>
<span id="cb131-24"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb131-24" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb132-1" tabindex="-1"></a><span class="fu">make_volcano</span>(<span class="fu">as.data.frame</span>(lfc_shrink),</span>
<span id="cb132-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb132-2" tabindex="-1"></a>             <span class="at">pval_nm=</span><span class="st">&quot;padj&quot;</span>, <span class="at">pval_cutoff=</span>fdr_cutoff)</span></code></pre></div>
<pre><code>## Warning: `aes_string()` was deprecated in ggplot2 3.0.0.
## ℹ Please use tidy evaluation idioms with `aes()`.
## ℹ See also `vignette(&quot;ggplot2-in-packages&quot;)` for more information.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_lifecycle_warnings()` to see where this warning was generated.</code></pre>
<p><img src="03_rnaseq_workflow_files/figure-html/volcano-1.png" width="384" /></p>
</div>
<div id="heatmap" class="section level3 hasAnchor" number="4.5.2">
<h3><span class="header-section-number">4.5.2</span> <strong>Heatmap</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#heatmap" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-1" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res)[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb134-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-2" tabindex="-1"></a></span>
<span id="cb134-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-3" tabindex="-1"></a>top_se <span class="ot">&lt;-</span> se[top_genes, ]</span>
<span id="cb134-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-4" tabindex="-1"></a>mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(top_se, <span class="st">&quot;vst&quot;</span>)</span>
<span id="cb134-5"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-5" tabindex="-1"></a>mat <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(mat), <span class="at">scale=</span><span class="cn">FALSE</span>, <span class="at">center =</span> <span class="cn">TRUE</span>))</span>
<span id="cb134-6"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-6" tabindex="-1"></a>  </span>
<span id="cb134-7"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-7" tabindex="-1"></a><span class="co"># column annot</span></span>
<span id="cb134-8"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-8" tabindex="-1"></a>ht_col_annot <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(top_se)[, <span class="st">&quot;group&quot;</span>, <span class="at">drop=</span><span class="cn">FALSE</span>])</span>
<span id="cb134-9"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-9" tabindex="-1"></a></span>
<span id="cb134-10"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-10" tabindex="-1"></a>group_lvls <span class="ot">&lt;-</span> <span class="fu">unique</span>(ht_col_annot<span class="sc">$</span>group)</span>
<span id="cb134-11"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-11" tabindex="-1"></a>ht_col_colors <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">group=</span><span class="fu">setNames</span>(<span class="fu">c</span>(<span class="st">&quot;#440154FF&quot;</span>,<span class="st">&quot;#2A788EFF&quot;</span>),</span>
<span id="cb134-12"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-12" tabindex="-1"></a>                                         <span class="at">nm=</span>group_lvls))</span>
<span id="cb134-13"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-13" tabindex="-1"></a></span>
<span id="cb134-14"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-14" tabindex="-1"></a><span class="fu">Heatmap</span>(mat, </span>
<span id="cb134-15"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-15" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">&quot;Mean-centered&quot;</span>, </span>
<span id="cb134-16"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-16" tabindex="-1"></a>        <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,</span>
<span id="cb134-17"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-17" tabindex="-1"></a>        <span class="at">row_labels=</span><span class="fu">rowData</span>(top_se)<span class="sc">$</span>Uniq_syms,</span>
<span id="cb134-18"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-18" tabindex="-1"></a>        <span class="at">show_column_names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb134-19"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-19" tabindex="-1"></a>        <span class="at">top_annotation=</span><span class="fu">HeatmapAnnotation</span>(<span class="at">df=</span>ht_col_annot, </span>
<span id="cb134-20"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-20" tabindex="-1"></a>                                         <span class="at">col=</span>ht_col_colors),</span>
<span id="cb134-21"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-21" tabindex="-1"></a>        <span class="at">column_title =</span> <span class="st">&quot;Top DE genes&quot;</span>,</span>
<span id="cb134-22"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-22" tabindex="-1"></a>        <span class="at">row_title =</span> <span class="fu">paste0</span>(<span class="fu">nrow</span>(mat), <span class="st">&quot; genes&quot;</span>)</span>
<span id="cb134-23"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb134-23" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="03_rnaseq_workflow_files/figure-html/heatmap-1.png" width="672" /></p>
</div>
<div id="p-value-distribution" class="section level3 hasAnchor" number="4.5.3">
<h3><span class="header-section-number">4.5.3</span> <strong>P value distribution</strong><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#p-value-distribution" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Ideally, we will see an anti-conservative (if there are many DE genes) or uniform pattern (not many DE genes). See <a href="http://varianceexplained.org/statistics/interpreting-pvalue-histogram/">here</a> for more details about how to interpret these.</p>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-1" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> <span class="fu">as.data.frame</span>(lfc_shrink) <span class="sc">%&gt;%</span></span>
<span id="cb135-2"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-2" tabindex="-1"></a>         dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(pvalue)),</span>
<span id="cb135-3"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-3" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> pvalue)) <span class="sc">+</span></span>
<span id="cb135-4"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-4" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;gray55&quot;</span>,</span>
<span id="cb135-5"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#cb135-5" tabindex="-1"></a>                 <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="fl">0.05</span>)) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.title=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>))</span></code></pre></div>
<p><img src="03_rnaseq_workflow_files/figure-html/pval-1.png" width="384" /></p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="mini-project.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="pathway-enrichment-and-experimental-design.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="assets/gitbook-2.6.7/js/app.min.js"></script>
<script src="assets/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="assets/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
},
"toolbar": {
"position": "static"
}
});
});
</script>

</body>

</html>
