<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Part 5 Pathway, enrichment, and experimental design | Introductory Bioinformatics for VAI Users</title>
  <meta name="description" content="Bioinformatics tutorials with hands-on exercises and examples." />
  <meta name="generator" content="bookdown 0.38 and GitBook 2.6.7" />

  <meta property="og:title" content="Part 5 Pathway, enrichment, and experimental design | Introductory Bioinformatics for VAI Users" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Bioinformatics tutorials with hands-on exercises and examples." />
  <meta name="github-repo" content="vari-bbc/Intro_to_Linux_for_HPC" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Part 5 Pathway, enrichment, and experimental design | Introductory Bioinformatics for VAI Users" />
  
  <meta name="twitter:description" content="Bioinformatics tutorials with hands-on exercises and examples." />
  

<meta name="author" content="The Bioinformatics and Biostatistics Core (BBC)" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"/>
<link rel="next" href="appendices.html"/>
<script src="assets/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="assets/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="assets/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="assets/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="assets/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="assets/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="assets/kePrint-0.0.1/kePrint.js"></script>
<link href="assets/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> <strong>Introduction</strong></a></li>
<li class="chapter" data-level="2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html"><i class="fa fa-check"></i><b>2</b> <strong>Basic Command Line Utilities on the HPC</strong></a>
<ul>
<li class="chapter" data-level="2.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#access-hpc"><i class="fa fa-check"></i><b>2.1</b> <strong>Access HPC</strong></a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#ondemand-instructions"><i class="fa fa-check"></i><b>2.1.1</b> <strong>OnDemand Instructions</strong></a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#file-navigation"><i class="fa fa-check"></i><b>2.2</b> <strong>File navigation</strong></a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#navigate-to-a-directory-folder"><i class="fa fa-check"></i><b>2.2.1</b> <strong>Navigate to a directory (folder)</strong></a></li>
<li class="chapter" data-level="2.2.2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#list-content-in-a-directory"><i class="fa fa-check"></i><b>2.2.2</b> <strong>List content in a directory</strong></a></li>
<li class="chapter" data-level="2.2.3" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#navigate-back-to-home-directory"><i class="fa fa-check"></i><b>2.2.3</b> <strong>Navigate back to home directory</strong></a></li>
<li class="chapter" data-level="2.2.4" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#create-a-directory"><i class="fa fa-check"></i><b>2.2.4</b> <strong>Create a directory</strong></a></li>
<li class="chapter" data-level="2.2.5" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#copy-a-file"><i class="fa fa-check"></i><b>2.2.5</b> <strong>Copy a file</strong></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#view-and-manipulate-files"><i class="fa fa-check"></i><b>2.3</b> <strong>View and manipulate files</strong></a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#display-content"><i class="fa fa-check"></i><b>2.3.1</b> <strong>Display content</strong></a></li>
<li class="chapter" data-level="2.3.2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#pattern-search"><i class="fa fa-check"></i><b>2.3.2</b> <strong>Pattern search</strong></a></li>
<li class="chapter" data-level="2.3.3" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#display-the-number-of-lines-words-and-characters"><i class="fa fa-check"></i><b>2.3.3</b> <strong>Display the number of lines, words, and characters</strong></a></li>
<li class="chapter" data-level="2.3.4" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#pipe---redirection"><i class="fa fa-check"></i><b>2.3.4</b> <strong>Pipe - redirection</strong></a></li>
<li class="chapter" data-level="2.3.5" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#output-redirection"><i class="fa fa-check"></i><b>2.3.5</b> <strong>Output redirection</strong></a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#exercise"><i class="fa fa-check"></i><b>2.4</b> <strong>Exercise</strong></a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#step-1-copy-the-files"><i class="fa fa-check"></i><b>2.4.1</b> <strong>Step 1: Copy the files</strong></a></li>
<li class="chapter" data-level="2.4.2" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#step-2-combine-the-files"><i class="fa fa-check"></i><b>2.4.2</b> <strong>Step 2: Combine the files</strong></a></li>
<li class="chapter" data-level="2.4.3" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#step-3-count-the-number-of-reads"><i class="fa fa-check"></i><b>2.4.3</b> <strong>Step 3: Count the number of reads</strong></a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="basic-command-line-utilities-on-the-hpc.html"><a href="basic-command-line-utilities-on-the-hpc.html#summary"><i class="fa fa-check"></i><b>2.5</b> <strong>Summary</strong></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="mini-project.html"><a href="mini-project.html"><i class="fa fa-check"></i><b>3</b> <strong>Mini Project</strong></a>
<ul>
<li class="chapter" data-level="3.1" data-path="mini-project.html"><a href="mini-project.html#start-an-interactive-job"><i class="fa fa-check"></i><b>3.1</b> <strong>Start an Interactive Job</strong></a></li>
<li class="chapter" data-level="3.2" data-path="mini-project.html"><a href="mini-project.html#create-directory"><i class="fa fa-check"></i><b>3.2</b> <strong>Create Directory</strong></a></li>
<li class="chapter" data-level="3.3" data-path="mini-project.html"><a href="mini-project.html#copy-fastq-files-to-fastqs-subdirectory"><i class="fa fa-check"></i><b>3.3</b> <strong>Copy Fastq Files To <code>fastqs</code> subdirectory</strong></a></li>
<li class="chapter" data-level="3.4" data-path="mini-project.html"><a href="mini-project.html#use-basic-linux-commands-to-explore-the-fastq-files"><i class="fa fa-check"></i><b>3.4</b> <strong>Use basic Linux commands to explore the fastq files</strong></a></li>
<li class="chapter" data-level="3.5" data-path="mini-project.html"><a href="mini-project.html#working-with-environment-modules"><i class="fa fa-check"></i><b>3.5</b> <strong>Working With Environment Modules</strong></a></li>
<li class="chapter" data-level="3.6" data-path="mini-project.html"><a href="mini-project.html#run-fastqc"><i class="fa fa-check"></i><b>3.6</b> <strong>Run FastQC</strong></a></li>
<li class="chapter" data-level="3.7" data-path="mini-project.html"><a href="mini-project.html#set-up-a-job-script-to-run-salmon-to-align-the-reads"><i class="fa fa-check"></i><b>3.7</b> <strong>Set up a job script to run Salmon to align the reads</strong></a></li>
<li class="chapter" data-level="3.8" data-path="mini-project.html"><a href="mini-project.html#submit-a-job"><i class="fa fa-check"></i><b>3.8</b> <strong>Submit a Job</strong></a></li>
<li class="chapter" data-level="3.9" data-path="mini-project.html"><a href="mini-project.html#examine-the-standard-output-stdout-and-standard-error-stderr-log-files"><i class="fa fa-check"></i><b>3.9</b> <strong>Examine the Standard Output (stdout) and Standard Error (stderr) log files</strong></a></li>
<li class="chapter" data-level="3.10" data-path="mini-project.html"><a href="mini-project.html#use-grep-to-find-the-tpms-for-a-specific-gene"><i class="fa fa-check"></i><b>3.10</b> <strong>Use grep to find the TPMs for a specific gene</strong></a></li>
<li class="chapter" data-level="3.11" data-path="mini-project.html"><a href="mini-project.html#bonus-use-an-interactive-job-to-run-multiqc-on-the-salmon-and-fastqc-output"><i class="fa fa-check"></i><b>3.11</b> <strong>BONUS: Use an interactive job to run multiQC on the Salmon and FastQC output</strong></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><i class="fa fa-check"></i><b>4</b> <strong>Basic RNA-Seq Analysis – From fastq files to DE genes</strong></a>
<ul>
<li class="chapter" data-level="4.1" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#run-the-bbc-pipeline-to-align-the-reads"><i class="fa fa-check"></i><b>4.1</b> <strong>Run the BBC pipeline to align the reads</strong></a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#the-workflow-as-a-graph"><i class="fa fa-check"></i><b>4.1.1</b> <strong>The workflow as a graph</strong></a></li>
<li class="chapter" data-level="4.1.2" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#clone-the-github-repo"><i class="fa fa-check"></i><b>4.1.2</b> <strong>Clone the Github repo</strong></a></li>
<li class="chapter" data-level="4.1.3" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#add-your-fastq-files-to-raw_data"><i class="fa fa-check"></i><b>4.1.3</b> <strong>Add your fastq files to <code>raw_data/</code></strong></a></li>
<li class="chapter" data-level="4.1.4" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#fill-out-the-samplesheet"><i class="fa fa-check"></i><b>4.1.4</b> <strong>Fill out the samplesheet</strong></a></li>
<li class="chapter" data-level="4.1.5" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#fill-out-the-config-file"><i class="fa fa-check"></i><b>4.1.5</b> <strong>Fill out the config file</strong></a></li>
<li class="chapter" data-level="4.1.6" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#submit-the-main-snakemake-job"><i class="fa fa-check"></i><b>4.1.6</b> <strong>Submit the main Snakemake job</strong></a></li>
<li class="chapter" data-level="4.1.7" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#bbc-maintained-reference-files"><i class="fa fa-check"></i><b>4.1.7</b> <strong>BBC-maintained reference files</strong></a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#quality-control"><i class="fa fa-check"></i><b>4.2</b> <strong>Quality control</strong></a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#multiqc-report"><i class="fa fa-check"></i><b>4.2.1</b> <strong>multiQC report</strong></a></li>
<li class="chapter" data-level="4.2.2" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#other-considerations"><i class="fa fa-check"></i><b>4.2.2</b> <strong>Other considerations</strong></a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#exploring-the-data-using-isee"><i class="fa fa-check"></i><b>4.3</b> <strong>Exploring the data using iSEE</strong></a></li>
<li class="chapter" data-level="4.4" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#de-analysis-using-r-and-deseq2"><i class="fa fa-check"></i><b>4.4</b> <strong>DE analysis using R and DESeq2</strong></a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#make-an-output-directory"><i class="fa fa-check"></i><b>4.4.1</b> <strong>Make an output directory</strong></a></li>
<li class="chapter" data-level="4.4.2" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#load-packages"><i class="fa fa-check"></i><b>4.4.2</b> <strong>Load packages</strong></a></li>
<li class="chapter" data-level="4.4.3" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#set-up-your-deseq-object"><i class="fa fa-check"></i><b>4.4.3</b> <strong>Set up your DESeq object</strong></a></li>
<li class="chapter" data-level="4.4.4" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#remove-genes-with-lowno-expression"><i class="fa fa-check"></i><b>4.4.4</b> <strong>Remove genes with low/no expression</strong></a></li>
<li class="chapter" data-level="4.4.5" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#different-normalization-approaches-for-different-biases"><i class="fa fa-check"></i><b>4.4.5</b> <strong>Different normalization approaches for different biases</strong></a></li>
<li class="chapter" data-level="4.4.6" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#run-the-de-workflow"><i class="fa fa-check"></i><b>4.4.6</b> <strong>Run the DE workflow</strong></a></li>
<li class="chapter" data-level="4.4.7" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#summarize-de-results"><i class="fa fa-check"></i><b>4.4.7</b> <strong>Summarize DE results</strong></a></li>
<li class="chapter" data-level="4.4.8" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#shrink-log-fold-changes-for-lowly-expressed-genes"><i class="fa fa-check"></i><b>4.4.8</b> <strong>Shrink log fold changes for lowly expressed genes</strong></a></li>
<li class="chapter" data-level="4.4.9" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#output-de-results"><i class="fa fa-check"></i><b>4.4.9</b> <strong>Output DE results</strong></a></li>
<li class="chapter" data-level="4.4.10" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#look-for-specific-genes"><i class="fa fa-check"></i><b>4.4.10</b> <strong>Look for specific genes</strong></a></li>
<li class="chapter" data-level="4.4.11" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#output-tables-with-raw-counts"><i class="fa fa-check"></i><b>4.4.11</b> <strong>Output tables with raw counts</strong></a></li>
<li class="chapter" data-level="4.4.12" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#output-tables-with-log2-normalized-counts"><i class="fa fa-check"></i><b>4.4.12</b> <strong>Output tables with log2 normalized counts</strong></a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#some-common-plots-for-deg-analysis"><i class="fa fa-check"></i><b>4.5</b> <strong>Some common plots for DEG analysis</strong></a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#volcano-plot"><i class="fa fa-check"></i><b>4.5.1</b> <strong>Volcano plot</strong></a></li>
<li class="chapter" data-level="4.5.2" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#heatmap"><i class="fa fa-check"></i><b>4.5.2</b> <strong>Heatmap</strong></a></li>
<li class="chapter" data-level="4.5.3" data-path="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html"><a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html#p-value-distribution"><i class="fa fa-check"></i><b>4.5.3</b> <strong>P value distribution</strong></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="pathway-enrichment-and-experimental-design.html"><a href="pathway-enrichment-and-experimental-design.html"><i class="fa fa-check"></i><b>5</b> <strong>Pathway, enrichment, and experimental design</strong></a>
<ul>
<li class="chapter" data-level="5.1" data-path="pathway-enrichment-and-experimental-design.html"><a href="pathway-enrichment-and-experimental-design.html#downstream-of-de-gene-analysis"><i class="fa fa-check"></i><b>5.1</b> <strong>Downstream of DE gene analysis</strong></a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="pathway-enrichment-and-experimental-design.html"><a href="pathway-enrichment-and-experimental-design.html#create-a-new-project-directory-for-enrichment-analysis"><i class="fa fa-check"></i><b>5.1.1</b> <strong>Create a new project directory for enrichment analysis</strong></a></li>
<li class="chapter" data-level="5.1.2" data-path="pathway-enrichment-and-experimental-design.html"><a href="pathway-enrichment-and-experimental-design.html#create-a-r-project"><i class="fa fa-check"></i><b>5.1.2</b> <strong>Create a R project</strong></a></li>
<li class="chapter" data-level="5.1.3" data-path="pathway-enrichment-and-experimental-design.html"><a href="pathway-enrichment-and-experimental-design.html#load-packaages"><i class="fa fa-check"></i><b>5.1.3</b> load packaages</a></li>
<li class="chapter" data-level="5.1.4" data-path="pathway-enrichment-and-experimental-design.html"><a href="pathway-enrichment-and-experimental-design.html#import-previous-de-results"><i class="fa fa-check"></i><b>5.1.4</b> Import previous DE results</a></li>
<li class="chapter" data-level="5.1.5" data-path="pathway-enrichment-and-experimental-design.html"><a href="pathway-enrichment-and-experimental-design.html#how-many-de-genes"><i class="fa fa-check"></i><b>5.1.5</b> How many DE genes</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="pathway-enrichment-and-experimental-design.html"><a href="pathway-enrichment-and-experimental-design.html#why-enrichment-and-pathway-analysis"><i class="fa fa-check"></i><b>5.2</b> Why enrichment and pathway analysis?</a></li>
<li class="chapter" data-level="5.3" data-path="pathway-enrichment-and-experimental-design.html"><a href="pathway-enrichment-and-experimental-design.html#ora"><i class="fa fa-check"></i><b>5.3</b> ORA</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="pathway-enrichment-and-experimental-design.html"><a href="pathway-enrichment-and-experimental-design.html#plot-ora-results"><i class="fa fa-check"></i><b>5.3.1</b> plot ORA results</a></li>
<li class="chapter" data-level="5.3.2" data-path="pathway-enrichment-and-experimental-design.html"><a href="pathway-enrichment-and-experimental-design.html#exercise-on-your-own"><i class="fa fa-check"></i><b>5.3.2</b> Exercise on your own</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="pathway-enrichment-and-experimental-design.html"><a href="pathway-enrichment-and-experimental-design.html#gsea"><i class="fa fa-check"></i><b>5.4</b> GSEA</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="pathway-enrichment-and-experimental-design.html"><a href="pathway-enrichment-and-experimental-design.html#plot-gsea-results"><i class="fa fa-check"></i><b>5.4.1</b> plot GSEA results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="appendices.html"><a href="appendices.html"><i class="fa fa-check"></i><b>6</b> <strong>Appendices</strong></a>
<ul>
<li class="chapter" data-level="6.1" data-path="appendices.html"><a href="appendices.html#workshop-powerpoint-files"><i class="fa fa-check"></i><b>6.1</b> <strong>Workshop Powerpoint files</strong></a></li>
<li class="chapter" data-level="6.2" data-path="appendices.html"><a href="appendices.html#local-access-of-files-on-the-hpc"><i class="fa fa-check"></i><b>6.2</b> <strong>Local access of files on the HPC</strong></a></li>
<li class="chapter" data-level="6.3" data-path="appendices.html"><a href="appendices.html#bash-cheatsheet"><i class="fa fa-check"></i><b>6.3</b> <strong>Bash cheatsheet</strong></a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introductory Bioinformatics for VAI Users</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="pathway-enrichment-and-experimental-design" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Part 5</span> <strong>Pathway, enrichment, and experimental design</strong><a href="pathway-enrichment-and-experimental-design.html#pathway-enrichment-and-experimental-design" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="downstream-of-de-gene-analysis" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> <strong>Downstream of DE gene analysis</strong><a href="pathway-enrichment-and-experimental-design.html#downstream-of-de-gene-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="create-a-new-project-directory-for-enrichment-analysis" class="section level3 hasAnchor" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> <strong>Create a new project directory for enrichment analysis</strong><a href="pathway-enrichment-and-experimental-design.html#create-a-new-project-directory-for-enrichment-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>You will make a new directory named “enrichment”, which is under “rnaseq_workflow”, and then you will create a R project in the “enrichment” folder.
First, located where you run your RNA-seq workflow and cd to that dir.</p>
<div class="sourceCode" id="cb133"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb133-1"><a href="pathway-enrichment-and-experimental-design.html#cb133-1" tabindex="-1"></a></span>
<span id="cb133-2"><a href="pathway-enrichment-and-experimental-design.html#cb133-2" tabindex="-1"></a><span class="bu">cd</span> /varidata/researchtemp/hpctmp/BBC_workshop_Oct2024_II/<span class="op">&lt;</span>user.name<span class="op">&gt;</span>/rnaseq_workflow</span>
<span id="cb133-3"><a href="pathway-enrichment-and-experimental-design.html#cb133-3" tabindex="-1"></a></span>
<span id="cb133-4"><a href="pathway-enrichment-and-experimental-design.html#cb133-4" tabindex="-1"></a><span class="fu">mkdir</span> enrichment</span>
<span id="cb133-5"><a href="pathway-enrichment-and-experimental-design.html#cb133-5" tabindex="-1"></a></span>
<span id="cb133-6"><a href="pathway-enrichment-and-experimental-design.html#cb133-6" tabindex="-1"></a><span class="bu">cd</span> enrichment</span>
<span id="cb133-7"><a href="pathway-enrichment-and-experimental-design.html#cb133-7" tabindex="-1"></a></span>
<span id="cb133-8"><a href="pathway-enrichment-and-experimental-design.html#cb133-8" tabindex="-1"></a><span class="bu">pwd</span>  <span class="co">## make sure you are in the right dir. </span></span></code></pre></div>
</div>
<div id="create-a-r-project" class="section level3 hasAnchor" number="5.1.2">
<h3><span class="header-section-number">5.1.2</span> <strong>Create a R project</strong><a href="pathway-enrichment-and-experimental-design.html#create-a-r-project" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>If you have are not in RStudio yet:
1. Please go to the following URL:<a href="https://ondemand3.vai.zone/">https://ondemand3.vai.zone/</a>.<br />
2. Click on RStudio Server, and create a new RStudio session on OnDemand. Please make sure you are loading the right R version, bbc2/R/alt/R-4.4.0-setR_LIBS_USER, which you can select from the drop down menu. Partition can be “short”, “long”, or “quick”, or “big”; Set CPU as 2, memory as 32, and time as 8 hours.<br />
3. In the Rstudio window, click “file” at the upper left corner, click on “New Project” -&gt; “New Directory” -&gt; “New Project”; Then click on “Browser”, and see if you can find your own rnaseq_workflow folder; if you can not find it, you can also click on triple dot horizontal button and manually type “/varidata/researchtemp/hpctmp/BBC_workshop_Oct2024_II” (make sure no typo), try to find your own folder then “rnaseq_workflow”, click on “choose”. Next you will go back to the project wizard, and type “enrichment”. Now, you should have created a R project under the rnaseq_workflow in the folder “enrichment”.
4. Lastly, you can create a R script file to put all your code for analysis there.</p>
<details>
<summary>
<strong>Click to expand</strong>Why create R project?
</summary>
<blockquote>
Creating an R Project in RStudio is recommended for several reasons:
1. A R Project creates a dedicated working directory for your project files, including R scripts, data files, figures, and output. This helps in org anise your files related to your specific project.<br />
2. Automatic Working Directory Setup. When you open an R Project, RStudio automatically sets the working directory to the project folder.<br />
3. Reproducibility. With a project-based setup, the code and files related to a project stay together, making it easier to reproduce results.
4. Version control integration and package development support.
</blockquote>
</details>
</div>
<div id="load-packaages" class="section level3 hasAnchor" number="5.1.3">
<h3><span class="header-section-number">5.1.3</span> load packaages<a href="pathway-enrichment-and-experimental-design.html#load-packaages" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="pathway-enrichment-and-experimental-design.html#cb134-1" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb134-2"><a href="pathway-enrichment-and-experimental-design.html#cb134-2" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb134-3"><a href="pathway-enrichment-and-experimental-design.html#cb134-3" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb134-4"><a href="pathway-enrichment-and-experimental-design.html#cb134-4" tabindex="-1"></a><span class="fu">library</span>(msigdbr)</span>
<span id="cb134-5"><a href="pathway-enrichment-and-experimental-design.html#cb134-5" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb134-6"><a href="pathway-enrichment-and-experimental-design.html#cb134-6" tabindex="-1"></a><span class="fu">library</span>(ggvenn)</span>
<span id="cb134-7"><a href="pathway-enrichment-and-experimental-design.html#cb134-7" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb134-8"><a href="pathway-enrichment-and-experimental-design.html#cb134-8" tabindex="-1"></a><span class="fu">library</span>(enrichplot)</span>
<span id="cb134-9"><a href="pathway-enrichment-and-experimental-design.html#cb134-9" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb134-10"><a href="pathway-enrichment-and-experimental-design.html#cb134-10" tabindex="-1"></a><span class="fu">library</span>(pathview)</span>
<span id="cb134-11"><a href="pathway-enrichment-and-experimental-design.html#cb134-11" tabindex="-1"></a>})</span></code></pre></div>
<pre><code>## Rows: 21485 Columns: 10
## ── Column specification ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
## Delimiter: &quot;\t&quot;
## chr (4): ens_gene, Symbol, Uniq_syms, Gene_name
## dbl (6): entrez, baseMean, log2FoldChange, lfcSE, pvalue, padj
## 
## ℹ Use `spec()` to retrieve the full column specification for this data.
## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<pre><code>## Warning in dir.create(outdir, recursive = TRUE): &#39;./enrichment_res&#39; already
## exists</code></pre>
</div>
<div id="import-previous-de-results" class="section level3 hasAnchor" number="5.1.4">
<h3><span class="header-section-number">5.1.4</span> Import previous DE results<a href="pathway-enrichment-and-experimental-design.html#import-previous-de-results" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="pathway-enrichment-and-experimental-design.html#cb137-1" tabindex="-1"></a>de_res <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(<span class="st">&quot;../DE_genes/deseq2_out_files/de_res.tsv&quot;</span>)</span></code></pre></div>
</div>
<div id="how-many-de-genes" class="section level3 hasAnchor" number="5.1.5">
<h3><span class="header-section-number">5.1.5</span> How many DE genes<a href="pathway-enrichment-and-experimental-design.html#how-many-de-genes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="pathway-enrichment-and-experimental-design.html#cb138-1" tabindex="-1"></a><span class="do">## note the part after the &quot;.&quot; sign is the version number, here we can just remove it.</span></span>
<span id="cb138-2"><a href="pathway-enrichment-and-experimental-design.html#cb138-2" tabindex="-1"></a>de_res<span class="sc">$</span>ens_gene <span class="ot">&lt;-</span> <span class="fu">str_split_fixed</span>(de_res<span class="sc">$</span>ens_gene, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>, <span class="dv">2</span>)[, <span class="dv">1</span>]</span>
<span id="cb138-3"><a href="pathway-enrichment-and-experimental-design.html#cb138-3" tabindex="-1"></a></span>
<span id="cb138-4"><a href="pathway-enrichment-and-experimental-design.html#cb138-4" tabindex="-1"></a><span class="do">## check the first a few rows again</span></span>
<span id="cb138-5"><a href="pathway-enrichment-and-experimental-design.html#cb138-5" tabindex="-1"></a><span class="fu">head</span>(de_res)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 10
##   ens_gene      Symbol Uniq_syms entrez Gene_name baseMean log2FoldChange  lfcSE
##   &lt;chr&gt;         &lt;chr&gt;  &lt;chr&gt;      &lt;dbl&gt; &lt;chr&gt;        &lt;dbl&gt;          &lt;dbl&gt;  &lt;dbl&gt;
## 1 ENSG00000152… SPARC… SPARCL1     8404 SPARC li…     994.           4.54 0.210 
## 2 ENSG00000148… STOM   STOM        2040 stomatin    13368.           1.41 0.0881
## 3 ENSG00000179… PER1   PER1        5187 period c…     764.           3.13 0.204 
## 4 ENSG00000134… PHC2   PHC2        1912 polyhome…    2708.           1.37 0.0902
## 5 ENSG00000120… DUSP1  DUSP1       1843 dual spe…    3357.           2.90 0.198 
## 6 ENSG00000125… MT2A   MT2A        4502 metallot…    3627.           2.17 0.150 
## # ℹ 2 more variables: pvalue &lt;dbl&gt;, padj &lt;dbl&gt;</code></pre>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="pathway-enrichment-and-experimental-design.html#cb140-1" tabindex="-1"></a>fdr_cutoff <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb140-2"><a href="pathway-enrichment-and-experimental-design.html#cb140-2" tabindex="-1"></a>up_reg_genes <span class="ot">&lt;-</span> de_res <span class="sc">|&gt;</span></span>
<span id="cb140-3"><a href="pathway-enrichment-and-experimental-design.html#cb140-3" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> fdr_cutoff <span class="sc">&amp;</span> log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span> )  </span>
<span id="cb140-4"><a href="pathway-enrichment-and-experimental-design.html#cb140-4" tabindex="-1"></a>down_reg_genes <span class="ot">&lt;-</span> de_res <span class="sc">|&gt;</span></span>
<span id="cb140-5"><a href="pathway-enrichment-and-experimental-design.html#cb140-5" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;</span> fdr_cutoff <span class="sc">&amp;</span> log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span> )  </span>
<span id="cb140-6"><a href="pathway-enrichment-and-experimental-design.html#cb140-6" tabindex="-1"></a></span>
<span id="cb140-7"><a href="pathway-enrichment-and-experimental-design.html#cb140-7" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Number of up-regulated genes is: &quot;</span>, <span class="fu">nrow</span>(up_reg_genes))</span></code></pre></div>
<pre><code>## Number of up-regulated genes is: 1872</code></pre>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="pathway-enrichment-and-experimental-design.html#cb142-1" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Number of down-regulated genes is: &quot;</span>, <span class="fu">nrow</span>(down_reg_genes))</span></code></pre></div>
<pre><code>## Number of down-regulated genes is: 1481</code></pre>
</div>
</div>
<div id="why-enrichment-and-pathway-analysis" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Why enrichment and pathway analysis?<a href="pathway-enrichment-and-experimental-design.html#why-enrichment-and-pathway-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>It can be difficult to look at so many genes as sometimes a list of genes do not provide any biological insights. Enrichment and pathway analyses are performed to determine if there is any konwn biological functions, interactions or pathways in your DE gene studies. Generally speaking, Enrichment and pathway analyses are restricted for DE gene expression, they can be performed on protemics data, ChIP-seq, and genomic data as well. There are different algorithms for enrichment and pathway analyses. Today, we will be focus on two types: ORA (over representation analysis) and GSEA (gene set enrichment analysis).</p>
</div>
<div id="ora" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> ORA<a href="pathway-enrichment-and-experimental-design.html#ora" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>ORA explores whether there is enrichment of known biological functions or pathways in a particular set of genes (e.g. significant up or down-regulated genes). Gene ontology (GO) is a framework used to represent and categorize gene functions across different species in a standardized manner. It provides a controlled vocabulary that describes genes in terms of their associated biological processes, cellular components, and molecular functions. The Gene Ontology Consortium maintains the GO vocabulary, as known as GO terms, and these GO terms are incorporated into gene annotations in many of the popular repositories for many organism.</p>
<details>
<summary>
<strong>Click to expand</strong>More details on GO:
</summary>
<blockquote>
<p>To describe the roles of genes and gene products, GO terms are organized into three independent components in a species-independent manner:</p>
<ul>
<li><p>Biological process: refers to the biological role involving the gene or gene product, and could include “transcription”, “signal transduction”, and “apoptosis”.</p></li>
<li><p>Molecular function: represents the biochemical activity of the gene product, such activities could include “ligand”, “protein binding”, and “enzyme activity”.</p></li>
<li><p>Cellular component: refers to the location in the cell of the gene product, such as “lysosome” and “plasma membrane”.</p></li>
<li><p>The GO terms are loosely hierarchical, ranging from general, ‘parent’, terms to more specific, ‘child’ terms.</p></li>
</ul>
</blockquote>
</details>
<p>Lastly, we are primarily using clusterProfiler for conducting enrichment analyses.</p>
<p>In the ORA here, we will be testing whether 2 fold up-regulated (logfC = 1, as logfold change is often log2 based) genes are enriched in which GO terms. We will be using enrichGO function from clusterProfiler, and use the whole dataset as the background (universe in enrichGO function).</p>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="pathway-enrichment-and-experimental-design.html#cb144-1" tabindex="-1"></a><span class="do">## only getting DE genes at least have 2 fold change, so log2FC &gt; 1, both up and down-regulated.</span></span>
<span id="cb144-2"><a href="pathway-enrichment-and-experimental-design.html#cb144-2" tabindex="-1"></a>genes_2f_up <span class="ot">&lt;-</span> de_res <span class="sc">%&gt;%</span> <span class="fu">filter</span>(padj <span class="sc">&lt;</span> fdr_cutoff <span class="sc">&amp;</span> log2FoldChange <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb144-3"><a href="pathway-enrichment-and-experimental-design.html#cb144-3" tabindex="-1"></a></span>
<span id="cb144-4"><a href="pathway-enrichment-and-experimental-design.html#cb144-4" tabindex="-1"></a><span class="fu">message</span>(<span class="st">&quot;Number of DE genes have at least 2 fold change is: &quot;</span>, <span class="fu">nrow</span>(genes_2f_up))</span></code></pre></div>
<pre><code>## Number of DE genes have at least 2 fold change is: 317</code></pre>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="pathway-enrichment-and-experimental-design.html#cb146-1" tabindex="-1"></a>ego_BP <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene          =</span> <span class="fu">as.character</span>(genes_2f_up<span class="sc">$</span>entrez),</span>
<span id="cb146-2"><a href="pathway-enrichment-and-experimental-design.html#cb146-2" tabindex="-1"></a>                <span class="at">universe      =</span> <span class="fu">as.character</span>(de_res<span class="sc">$</span>entrez),</span>
<span id="cb146-3"><a href="pathway-enrichment-and-experimental-design.html#cb146-3" tabindex="-1"></a>                <span class="at">OrgDb         =</span> org.Hs.eg.db,</span>
<span id="cb146-4"><a href="pathway-enrichment-and-experimental-design.html#cb146-4" tabindex="-1"></a>                <span class="at">ont           =</span> <span class="st">&quot;BP&quot;</span>,</span>
<span id="cb146-5"><a href="pathway-enrichment-and-experimental-design.html#cb146-5" tabindex="-1"></a>                <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb146-6"><a href="pathway-enrichment-and-experimental-design.html#cb146-6" tabindex="-1"></a>                <span class="at">qvalueCutoff  =</span> <span class="fl">0.05</span>,</span>
<span id="cb146-7"><a href="pathway-enrichment-and-experimental-design.html#cb146-7" tabindex="-1"></a>                <span class="at">keyType =</span> <span class="st">&quot;ENTREZID&quot;</span>,</span>
<span id="cb146-8"><a href="pathway-enrichment-and-experimental-design.html#cb146-8" tabindex="-1"></a>                <span class="at">readable =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div id="plot-ora-results" class="section level3 hasAnchor" number="5.3.1">
<h3><span class="header-section-number">5.3.1</span> plot ORA results<a href="pathway-enrichment-and-experimental-design.html#plot-ora-results" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Dotplots can be used to depict the enrichment scores (e.g. p values) and gene count or ratio as circle size and colors. Cneplots, gene-concept network, allows us to know which genes are involved in these significant terms or pathways, as well as the linkages among GO terms or pathways.</p>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="pathway-enrichment-and-experimental-design.html#cb147-1" tabindex="-1"></a><span class="fu">dotplot</span>(ego_BP, <span class="at">showCategory =</span> <span class="dv">20</span>, <span class="at">font.size =</span> <span class="dv">9</span> ) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;dotplot for ORA - BP&quot;</span>)</span></code></pre></div>
<p><img src="04_pathway_enrichment_files/figure-html/plot_ORA-1.svg" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="pathway-enrichment-and-experimental-design.html#cb148-1" tabindex="-1"></a><span class="fu">cnetplot</span>(ego_BP, <span class="at">showCategory =</span> <span class="dv">10</span>, <span class="at">node_label_size =</span> <span class="fl">0.6</span>, <span class="at">cex_label_category =</span> <span class="fl">0.8</span>,</span>
<span id="cb148-2"><a href="pathway-enrichment-and-experimental-design.html#cb148-2" tabindex="-1"></a>  <span class="at">cex_label_gene =</span> <span class="fl">0.5</span>, <span class="at">cex_category =</span> <span class="fl">0.8</span>, <span class="at">cex_genes =</span> <span class="fl">0.5</span>, </span>
<span id="cb148-3"><a href="pathway-enrichment-and-experimental-design.html#cb148-3" tabindex="-1"></a>  <span class="at">max.overlaps =</span> <span class="cn">Inf</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;Gene-concept network for ORA - GO:BP&quot;</span>)</span></code></pre></div>
<pre><code>## Warning in cnetplot.enrichResult(x, ...): Use &#39;cex.params = list(category_node = your_value)&#39; instead of &#39;cex_category&#39;.
##  The cex_category parameter will be removed in the next version.</code></pre>
<pre><code>## Warning in cnetplot.enrichResult(x, ...): Use &#39;cex.params = list(category_label = your_value)&#39; instead of &#39;cex_label_category&#39;.
##  The cex_label_category parameter will be removed in the next version.</code></pre>
<pre><code>## Warning in cnetplot.enrichResult(x, ...): Use &#39;cex.params = list(gene_label = your_value)&#39; instead of &#39;cex_label_gene&#39;.
##  The cex_label_gene parameter will be removed in the next version.</code></pre>
<pre><code>## Warning: ggrepel: 1 unlabeled data points (too many overlaps). Consider
## increasing max.overlaps</code></pre>
<p><img src="04_pathway_enrichment_files/figure-html/plot_ORA-2.svg" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="pathway-enrichment-and-experimental-design.html#cb153-1" tabindex="-1"></a>ego_BP_df <span class="ot">&lt;-</span> ego_BP <span class="sc">|&gt;</span> <span class="fu">as.data.frame</span>() </span>
<span id="cb153-2"><a href="pathway-enrichment-and-experimental-design.html#cb153-2" tabindex="-1"></a></span>
<span id="cb153-3"><a href="pathway-enrichment-and-experimental-design.html#cb153-3" tabindex="-1"></a><span class="do">## output the BP enrichment results into a tsv file</span></span>
<span id="cb153-4"><a href="pathway-enrichment-and-experimental-design.html#cb153-4" tabindex="-1"></a><span class="fu">write_tsv</span>(ego_BP_df, <span class="fu">paste0</span>(outdir, <span class="st">&quot;/clusterProfiler_BP_res.tsv&quot;</span>))</span></code></pre></div>
<details>
<summary>
<strong>Click to expand</strong>Exercise on your own:
</summary>
<blockquote>
</div>
<div id="exercise-on-your-own" class="section level3 hasAnchor" number="5.3.2">
<h3><span class="header-section-number">5.3.2</span> Exercise on your own<a href="pathway-enrichment-and-experimental-design.html#exercise-on-your-own" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p>Using the same conditions, return the enriched GO processes for the Molecular Function ontology.</p></li>
<li><p>How would the enrichGO() function change if our organism was rat?</p></li>
</ol>
</blockquote>
</details>
</div>
</div>
<div id="gsea" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> GSEA<a href="pathway-enrichment-and-experimental-design.html#gsea" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>the difference is small, but evidenced in coordinated way in a set of related genes. GSEA (Subramanian et al. 2005) directly addresses this limitation. Unlike ORA where an arbitary threshold is set as “significant DE genes”, all genes can be used in GSEA. We usually sort genes based on their fold changes or pvalues, and compare this sorted gene list to a predefined gene set, such as genes in certain pathways. It will help to detect situations where all genes in a predefined set change in a small but coordinated manner. P-values are generated through permutation tests. This type of analysis can be particularly helpful if the differential expression analysis only outputs a small number of significant DE genes.</p>
<p>Here, what we create is a sorted list of all the genes in our dataset sorted by -log10(pvalue) and sign of fold change; this way, up-regulated and down-regulated genes will be at the top and bottom of the list. In GSEA, we will show how to run GSEA on <a href="https://www.genome.jp/kegg/pathway.html">KEGG pathways</a></p>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="pathway-enrichment-and-experimental-design.html#cb154-1" tabindex="-1"></a><span class="do">## we will use all the genes. Get the fold change first</span></span>
<span id="cb154-2"><a href="pathway-enrichment-and-experimental-design.html#cb154-2" tabindex="-1"></a>gene_list <span class="ot">&lt;-</span> <span class="fu">sign</span>(de_res<span class="sc">$</span>log2FoldChange) <span class="sc">*</span> <span class="sc">-</span><span class="fu">log10</span>(de_res<span class="sc">$</span>pvalue)</span>
<span id="cb154-3"><a href="pathway-enrichment-and-experimental-design.html#cb154-3" tabindex="-1"></a><span class="do">## or we can just use fold change data. </span></span>
<span id="cb154-4"><a href="pathway-enrichment-and-experimental-design.html#cb154-4" tabindex="-1"></a></span>
<span id="cb154-5"><a href="pathway-enrichment-and-experimental-design.html#cb154-5" tabindex="-1"></a><span class="do">## get the name of the genes. </span></span>
<span id="cb154-6"><a href="pathway-enrichment-and-experimental-design.html#cb154-6" tabindex="-1"></a><span class="fu">names</span>(gene_list) <span class="ot">&lt;-</span> de_res<span class="sc">$</span>entrez <span class="do">## the names of the gene list is the en ID. </span></span>
<span id="cb154-7"><a href="pathway-enrichment-and-experimental-design.html#cb154-7" tabindex="-1"></a></span>
<span id="cb154-8"><a href="pathway-enrichment-and-experimental-design.html#cb154-8" tabindex="-1"></a><span class="do">## remove NA is there is any in the fold change or gene names</span></span>
<span id="cb154-9"><a href="pathway-enrichment-and-experimental-design.html#cb154-9" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">any</span>(<span class="fu">is.na</span>(gene_list))){ </span>
<span id="cb154-10"><a href="pathway-enrichment-and-experimental-design.html#cb154-10" tabindex="-1"></a>  gene_list <span class="ot">=</span> gene_list[<span class="sc">-</span><span class="fu">which</span>(<span class="fu">is.na</span>(gene_list))]</span>
<span id="cb154-11"><a href="pathway-enrichment-and-experimental-design.html#cb154-11" tabindex="-1"></a>}</span>
<span id="cb154-12"><a href="pathway-enrichment-and-experimental-design.html#cb154-12" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">any</span>(<span class="fu">is.na</span>(<span class="fu">names</span>(gene_list)))) {</span>
<span id="cb154-13"><a href="pathway-enrichment-and-experimental-design.html#cb154-13" tabindex="-1"></a>  gene_list <span class="ot">=</span> gene_list[<span class="sc">-</span><span class="fu">which</span>(<span class="fu">is.na</span>(<span class="fu">names</span>(gene_list)))]</span>
<span id="cb154-14"><a href="pathway-enrichment-and-experimental-design.html#cb154-14" tabindex="-1"></a>}</span>
<span id="cb154-15"><a href="pathway-enrichment-and-experimental-design.html#cb154-15" tabindex="-1"></a></span>
<span id="cb154-16"><a href="pathway-enrichment-and-experimental-design.html#cb154-16" tabindex="-1"></a><span class="do">## also remove duplicated gene names</span></span>
<span id="cb154-17"><a href="pathway-enrichment-and-experimental-design.html#cb154-17" tabindex="-1"></a>gene_list <span class="ot">&lt;-</span> gene_list[<span class="sc">!</span><span class="fu">duplicated</span>(<span class="fu">names</span>(gene_list))]</span>
<span id="cb154-18"><a href="pathway-enrichment-and-experimental-design.html#cb154-18" tabindex="-1"></a></span>
<span id="cb154-19"><a href="pathway-enrichment-and-experimental-design.html#cb154-19" tabindex="-1"></a><span class="do">## sort from largest to smallest.</span></span>
<span id="cb154-20"><a href="pathway-enrichment-and-experimental-design.html#cb154-20" tabindex="-1"></a>gene_list <span class="ot">&lt;-</span> <span class="fu">sort</span>(gene_list, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span>
<span id="cb154-21"><a href="pathway-enrichment-and-experimental-design.html#cb154-21" tabindex="-1"></a></span>
<span id="cb154-22"><a href="pathway-enrichment-and-experimental-design.html#cb154-22" tabindex="-1"></a><span class="fu">head</span>(gene_list)</span></code></pre></div>
<pre><code>##      8404      2040      5187      1912      1843      4502 
## 106.41851  59.25457  54.02435  53.41694  49.03381  48.37430</code></pre>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="pathway-enrichment-and-experimental-design.html#cb156-1" tabindex="-1"></a><span class="co"># set seeds</span></span>
<span id="cb156-2"><a href="pathway-enrichment-and-experimental-design.html#cb156-2" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1234</span>)</span>
<span id="cb156-3"><a href="pathway-enrichment-and-experimental-design.html#cb156-3" tabindex="-1"></a></span>
<span id="cb156-4"><a href="pathway-enrichment-and-experimental-design.html#cb156-4" tabindex="-1"></a><span class="do">## run GSEA using clusterProfiler</span></span>
<span id="cb156-5"><a href="pathway-enrichment-and-experimental-design.html#cb156-5" tabindex="-1"></a>gseaKEGG <span class="ot">&lt;-</span> <span class="fu">gseKEGG</span>(<span class="at">geneList =</span> gene_list, <span class="co"># ordered named vector of fold changes (Entrez IDs are the associated names)</span></span>
<span id="cb156-6"><a href="pathway-enrichment-and-experimental-design.html#cb156-6" tabindex="-1"></a>              <span class="at">organism =</span> <span class="st">&quot;hsa&quot;</span>, </span>
<span id="cb156-7"><a href="pathway-enrichment-and-experimental-design.html#cb156-7" tabindex="-1"></a>              <span class="at">nPerm =</span> <span class="dv">1000</span>, <span class="co"># default number permutations</span></span>
<span id="cb156-8"><a href="pathway-enrichment-and-experimental-design.html#cb156-8" tabindex="-1"></a>              <span class="at">minGSSize =</span> <span class="dv">100</span>, <span class="co"># minimum gene set size (# genes in set) - change to test more sets or recover sets with fewer # genes</span></span>
<span id="cb156-9"><a href="pathway-enrichment-and-experimental-design.html#cb156-9" tabindex="-1"></a>              <span class="at">pvalueCutoff =</span> <span class="fl">0.1</span>, <span class="co"># padj cutoff value</span></span>
<span id="cb156-10"><a href="pathway-enrichment-and-experimental-design.html#cb156-10" tabindex="-1"></a>              <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## Reading KEGG annotation online: &quot;https://rest.kegg.jp/link/hsa/pathway&quot;...</code></pre>
<pre><code>## Reading KEGG annotation online: &quot;https://rest.kegg.jp/list/pathway/hsa&quot;...</code></pre>
<pre><code>## Warning in .GSEA(geneList = geneList, exponent = exponent, minGSSize =
## minGSSize, : We do not recommend using nPerm parameter incurrent and future
## releases</code></pre>
<pre><code>## Warning in fgsea(pathways = geneSets, stats = geneList, nperm = nPerm, minSize
## = minGSSize, : You are trying to run fgseaSimple. It is recommended to use
## fgseaMultilevel. To run fgseaMultilevel, you need to remove the nperm argument
## in the fgsea function call.</code></pre>
<pre><code>## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (0.04% of the list).
## The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb162-1"><a href="pathway-enrichment-and-experimental-design.html#cb162-1" tabindex="-1"></a>gseaKEGG_gene_symbol <span class="ot">&lt;-</span> <span class="fu">setReadable</span>(gseaKEGG, <span class="st">&#39;org.Hs.eg.db&#39;</span>, <span class="st">&#39;ENTREZID&#39;</span>)</span>
<span id="cb162-2"><a href="pathway-enrichment-and-experimental-design.html#cb162-2" tabindex="-1"></a></span>
<span id="cb162-3"><a href="pathway-enrichment-and-experimental-design.html#cb162-3" tabindex="-1"></a></span>
<span id="cb162-4"><a href="pathway-enrichment-and-experimental-design.html#cb162-4" tabindex="-1"></a><span class="do">### again, gsea is a large list, and we convert it to a data frame, and save the results</span></span>
<span id="cb162-5"><a href="pathway-enrichment-and-experimental-design.html#cb162-5" tabindex="-1"></a>gsea_df <span class="ot">&lt;-</span> gseaKEGG_gene_symbol <span class="sc">|&gt;</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb162-6"><a href="pathway-enrichment-and-experimental-design.html#cb162-6" tabindex="-1"></a></span>
<span id="cb162-7"><a href="pathway-enrichment-and-experimental-design.html#cb162-7" tabindex="-1"></a><span class="fu">write_tsv</span>(gsea_df, <span class="fu">paste0</span>(outdir, <span class="st">&quot;/clusterProfiler_KEGG_GSEA_res.tsv&quot;</span>))</span></code></pre></div>
<div id="plot-gsea-results" class="section level3 hasAnchor" number="5.4.1">
<h3><span class="header-section-number">5.4.1</span> plot GSEA results<a href="pathway-enrichment-and-experimental-design.html#plot-gsea-results" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Similarity, we can make dotplot and gene-concept network plots here.</p>
<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="pathway-enrichment-and-experimental-design.html#cb163-1" tabindex="-1"></a><span class="co"># first dotplot</span></span>
<span id="cb163-2"><a href="pathway-enrichment-and-experimental-design.html#cb163-2" tabindex="-1"></a><span class="fu">dotplot</span>(gseaKEGG_gene_symbol, <span class="at">showCategory =</span> <span class="dv">15</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;dotplot for gsea - KEGG pathways&quot;</span>)</span></code></pre></div>
<p><img src="04_pathway_enrichment_files/figure-html/plot_GSEA1-1.svg" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb164"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb164-1"><a href="pathway-enrichment-and-experimental-design.html#cb164-1" tabindex="-1"></a><span class="fu">cnetplot</span>(gseaKEGG_gene_symbol, </span>
<span id="cb164-2"><a href="pathway-enrichment-and-experimental-design.html#cb164-2" tabindex="-1"></a>         <span class="at">showCategory =</span> <span class="dv">5</span>, </span>
<span id="cb164-3"><a href="pathway-enrichment-and-experimental-design.html#cb164-3" tabindex="-1"></a>         <span class="at">node_label_size =</span> <span class="fl">0.6</span>, </span>
<span id="cb164-4"><a href="pathway-enrichment-and-experimental-design.html#cb164-4" tabindex="-1"></a>         <span class="at">cex_label_category =</span> <span class="fl">0.8</span>,</span>
<span id="cb164-5"><a href="pathway-enrichment-and-experimental-design.html#cb164-5" tabindex="-1"></a>         <span class="at">cex_label_gene =</span> <span class="fl">0.5</span>, </span>
<span id="cb164-6"><a href="pathway-enrichment-and-experimental-design.html#cb164-6" tabindex="-1"></a>         <span class="at">cex_category =</span> <span class="fl">0.8</span>, <span class="at">cex_genes =</span> <span class="fl">0.5</span>, </span>
<span id="cb164-7"><a href="pathway-enrichment-and-experimental-design.html#cb164-7" tabindex="-1"></a>         <span class="at">max.overlaps =</span> <span class="cn">Inf</span>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;Gene-concept network for gsea-KEGG pathways&quot;</span>)</span></code></pre></div>
<pre><code>## Warning in cnetplot.enrichResult(x, ...): Use &#39;cex.params = list(category_node = your_value)&#39; instead of &#39;cex_category&#39;.
##  The cex_category parameter will be removed in the next version.</code></pre>
<pre><code>## Warning in cnetplot.enrichResult(x, ...): Use &#39;cex.params = list(category_label = your_value)&#39; instead of &#39;cex_label_category&#39;.
##  The cex_label_category parameter will be removed in the next version.</code></pre>
<pre><code>## Warning in cnetplot.enrichResult(x, ...): Use &#39;cex.params = list(gene_label = your_value)&#39; instead of &#39;cex_label_gene&#39;.
##  The cex_label_gene parameter will be removed in the next version.</code></pre>
<pre><code>## Warning: ggrepel: 1 unlabeled data points (too many overlaps). Consider
## increasing max.overlaps</code></pre>
<p><img src="04_pathway_enrichment_files/figure-html/plot_GSEA1-2.svg" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb169"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb169-1"><a href="pathway-enrichment-and-experimental-design.html#cb169-1" tabindex="-1"></a><span class="co"># Specific plots we can make for GSEA results</span></span>
<span id="cb169-2"><a href="pathway-enrichment-and-experimental-design.html#cb169-2" tabindex="-1"></a><span class="fu">gseaplot</span>(gseaKEGG_gene_symbol, <span class="at">geneSetID =</span> <span class="st">&#39;hsa04910&#39;</span>)</span></code></pre></div>
<p><img src="04_pathway_enrichment_files/figure-html/plot_GSEA1-3.svg" width="768" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb170"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb170-1"><a href="pathway-enrichment-and-experimental-design.html#cb170-1" tabindex="-1"></a><span class="co"># Lastly, we can map gene expression to a KEGG map</span></span>
<span id="cb170-2"><a href="pathway-enrichment-and-experimental-design.html#cb170-2" tabindex="-1"></a>foldchanges <span class="ot">&lt;-</span> de_res<span class="sc">$</span>log2FoldChange</span>
<span id="cb170-3"><a href="pathway-enrichment-and-experimental-design.html#cb170-3" tabindex="-1"></a><span class="fu">names</span>(foldchanges) <span class="ot">&lt;-</span> de_res<span class="sc">$</span>entrez</span>
<span id="cb170-4"><a href="pathway-enrichment-and-experimental-design.html#cb170-4" tabindex="-1"></a></span>
<span id="cb170-5"><a href="pathway-enrichment-and-experimental-design.html#cb170-5" tabindex="-1"></a><span class="fu">pathview</span>(<span class="at">gene.data =</span> foldchanges,</span>
<span id="cb170-6"><a href="pathway-enrichment-and-experimental-design.html#cb170-6" tabindex="-1"></a>              <span class="at">pathway.id =</span> <span class="st">&#39;hsa04910&#39;</span>,</span>
<span id="cb170-7"><a href="pathway-enrichment-and-experimental-design.html#cb170-7" tabindex="-1"></a>              <span class="at">species =</span> <span class="st">&quot;hsa&quot;</span>,</span>
<span id="cb170-8"><a href="pathway-enrichment-and-experimental-design.html#cb170-8" tabindex="-1"></a>              <span class="at">limit =</span> <span class="fu">list</span>(<span class="at">gene =</span> <span class="dv">2</span>, <span class="co"># value gives the max/min limit for foldchanges</span></span>
<span id="cb170-9"><a href="pathway-enrichment-and-experimental-design.html#cb170-9" tabindex="-1"></a>              <span class="at">cpd =</span> <span class="dv">1</span>))</span>
<span id="cb170-10"><a href="pathway-enrichment-and-experimental-design.html#cb170-10" tabindex="-1"></a></span>
<span id="cb170-11"><a href="pathway-enrichment-and-experimental-design.html#cb170-11" tabindex="-1"></a><span class="do">## you can download the map from in Rstudio, but you will be seeing something like this. </span></span></code></pre></div>
<div class="float">
<img src="img/hsa04910.pathview.png" alt="Example pathview output." />
<div class="figcaption">Example pathview output.</div>
</div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="basic-rna-seq-analysis-from-fastq-files-to-de-genes.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="appendices.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="assets/gitbook-2.6.7/js/app.min.js"></script>
<script src="assets/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="assets/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="assets/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
},
"toolbar": {
"position": "static"
}
});
});
</script>

</body>

</html>
